--- dbg/find14/find/.cvsignore	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/.cvsignore	2020-01-28 23:39:42.000000000 -0600
@@ -3,11 +3,18 @@ Makefile.in
 aclocal.m4
 build
 config.cache
+autom4te.cache
 config.h
 config.h.in
 configure
 config.log
 config.status
-stamp-h
+stamp-h1
 config.sub
 config.rpath
+config.guess
+gnulib
+install-sh
+link-warning.h
+gnulib-cvs
+tests
Only in dbg/find6/find/debian: changelog
Only in dbg/find14/find/debian: Changelog
diff -up -r dbg/find14/find/debian/copyright dbg/find6/find/debian/copyright
