--- dbg/find14/find/gnulib/lib/__fpending.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/__fpending.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* __fpending.c -- return the number of pending output bytes on a stream
-   Copyright (C) 2000, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2000, 2004, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,13 +13,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by Jim Meyering. */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include "__fpending.h"
 
diff -up -r dbg/find14/find/gnulib/lib/__fpending.h dbg/find6/find/gnulib/lib/__fpending.h
