--- dbg/find14/find/gnulib/lib/__fpending.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/__fpending.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,17 +1,34 @@
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+/* Declare __fpending.
+
+   Copyright (C) 2000, 2003, 2005, 2006 Free Software Foundation, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2, or (at your option)
+   any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+   Written by Jim Meyering.  */
 
 #include <stddef.h>
 #include <stdio.h>
 
-#if HAVE_STDIO_EXT_H
-# include <stdio_ext.h>
-#endif
-
 #ifndef HAVE_DECL___FPENDING
 "this configure-time declaration test was not run"
 #endif
-#if !HAVE_DECL___FPENDING
+
+#if HAVE_DECL___FPENDING
+# if HAVE_STDIO_EXT_H
+#  include <stdio_ext.h>
+# endif
+#else
 size_t __fpending (FILE *);
 #endif
Only in dbg/find6/find/gnulib/lib: fstatat.c
Only in dbg/find6/find/gnulib/lib: fstatat.o
Only in dbg/find6/find/gnulib/lib: fts.c
Only in dbg/find6/find/gnulib/lib: fts-cycle.c
Only in dbg/find6/find/gnulib/lib: fts_.h
Only in dbg/find6/find/gnulib/lib: fts.o
diff -up -r dbg/find14/find/gnulib/lib/getcwd.c dbg/find6/find/gnulib/lib/getcwd.c
