--- dbg/find14/find/gnulib-cvs/gnulib/lib/acosl.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/acosl.c	2020-01-28 23:39:42.000000000 -0600
@@ -9,7 +9,10 @@
  * ====================================================
  */
 
-#include "mathl.h"
+#include <config.h>
+
+/* Specification.  */
+#include <math.h>
 
 /*
   Long double expansions contributed by
@@ -33,8 +36,6 @@
  */
 
 
-#include <math.h>
-
 static const long double
   one = 1.0L,
   huge = 1.0e+4932L,
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/alloca.c dbg/find6/find/gnulib-cvs/gnulib/lib/alloca.c
