--- dbg/find14/find/gnulib/lib/alloca.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/alloca.c	2020-01-28 23:39:42.000000000 -0600
@@ -21,9 +21,7 @@
    allocating any.  It is a good idea to use alloca(0) in
    your main control loop, etc. to force garbage collection.  */
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include <alloca.h>
 
diff -up -r dbg/find14/find/gnulib/lib/alloca_.h dbg/find6/find/gnulib/lib/alloca_.h
