--- dbg/find14/find/gnulib-cvs/gnulib/modules/alloca	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/alloca	2020-01-28 23:39:42.000000000 -0600
@@ -15,7 +15,7 @@ Include:
 #include <alloca.h>
 
 License:
-GPL
+public domain
 
 Maintainer:
 bug-gnulib@gnu.org
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/alloca-opt dbg/find6/find/gnulib-cvs/gnulib/modules/alloca-opt
