--- dbg/find14/find/gnulib-cvs/gnulib/modules/argmatch	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/argmatch	2020-01-28 23:39:42.000000000 -0600
@@ -4,19 +4,22 @@ Search for an exact or abbreviated match
 Files:
 lib/argmatch.h
 lib/argmatch.c
+m4/argmatch.m4
 
 Depends-on:
-gettext
+gettext-h
 error
 quotearg
 quote
 exit
 exitfail
+verify
+stdbool
 
 configure.ac:
+gl_ARGMATCH
 
 Makefile.am:
-lib_SOURCES += argmatch.h argmatch.c
 
 Include:
 "argmatch.h"
@@ -26,4 +29,3 @@ GPL
 
 Maintainer:
 all
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: argmatch-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/argp dbg/find6/find/gnulib-cvs/gnulib/modules/argp
