--- dbg/find14/find/gnulib-cvs/gnulib/modules/argp	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/argp	2020-01-28 23:39:42.000000000 -0600
@@ -11,6 +11,7 @@ lib/argp-fs-xinl.c
 lib/argp-help.c
 lib/argp-namefrob.h
 lib/argp-parse.c
+lib/argp-pin.c
 lib/argp-pv.c
 lib/argp-pvh.c
 lib/argp-xinl.c
@@ -18,12 +19,13 @@ m4/argp.m4
 
 Depends-on:
 alloca
+dirname
 getopt
 strchrnul
 sysexits
+malloc
 mempcpy
 strndup
-restrict
 strcase
 extensions
 vsnprintf
@@ -34,7 +36,8 @@ gl_ARGP
 Makefile.am:
 lib_SOURCES += argp.h argp-ba.c argp-eexst.c \
            argp-fmtstream.c argp-fmtstream.h argp-fs-xinl.c argp-help.c \
-           argp-namefrob.h argp-parse.c argp-pv.c argp-pvh.c argp-xinl.c
+           argp-namefrob.h argp-parse.c argp-pin.c argp-pv.c argp-pvh.c \
+           argp-xinl.c
 
 Include:
 "argp.h"
@@ -44,4 +47,3 @@ LGPL
 
 Maintainer:
 Sergey Poznyakoff, glibc
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: argp-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/argz dbg/find6/find/gnulib-cvs/gnulib/modules/argz
