--- dbg/find14/find/gnulib-cvs/gnulib/m4/argz.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/argz.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,13 +1,13 @@
 # Portability macros for glibc argz.                    -*- Autoconf -*-
-# Written by Gary V. Vaughan <gary@gnu.org>
-
-# Copyright (C) 2004, 2005 Free Software Foundation, Inc.
-
+#
+#   Copyright (C) 2004, 2005, 2006 Free Software Foundation, Inc.
+#   Written by Gary V. Vaughan <gary@gnu.org>
+#
 # This file is free software; the Free Software Foundation gives
 # unlimited permission to copy and/or distribute it, with or without
 # modifications, as long as this notice is preserved.
 
-# serial 2
+# serial 4 argz.m4
 
 AC_DEFUN([gl_FUNC_ARGZ],
 [gl_PREREQ_ARGZ
@@ -17,7 +17,9 @@ AC_CHECK_HEADERS([argz.h], [], [], [AC_I
 AC_CHECK_TYPES([error_t],
   [],
   [AC_DEFINE([error_t], [int],
-   [Define to a type to use for `error_t' if it is not otherwise available.])],
+   [Define to a type to use for `error_t' if it is not otherwise available.])
+   AC_DEFINE([__error_t_defined], [1], [Define so that glibc/gnulib argp.h
+    does not typedef error_t.])],
   [#if defined(HAVE_ARGZ_H)
 #  include <argz.h>
 #endif])
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: arpa_inet_h.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/backupfile.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/backupfile.m4
