--- dbg/find14/find/gnulib-cvs/gnulib/lib/argz_.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/argz_.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,23 +1,32 @@
 /* lt__argz.h -- internal argz interface for non-glibc systems
+
    Copyright (C) 2004 Free Software Foundation, Inc.
-   Originally by Gary V. Vaughan  <gary@gnu.org>
+   Written by Gary V. Vaughan, 2004
 
    NOTE: The canonical source of this file is maintained with the
    GNU Libtool package.  Report bugs to bug-libtool@gnu.org.
 
-   This program is free software; you can redistribute it and/or modify
-   it under the terms of the GNU General Public License as published by
-   the Free Software Foundation; either version 2, or (at your option)
-   any later version.
-
-   This program is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-   GNU General Public License for more details.
-
-   You should have received a copy of the GNU General Public License along
-   with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+GNU Libltdl is free software; you can redistribute it and/or
+modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation; either
+version 2 of the License, or (at your option) any later version.
+
+As a special exception to the GNU Lesser General Public License,
+if you distribute this file as part of a program or library that
+is built using GNU Libtool, you may include this file under the
+same distribution terms that you use for the rest of that program.
+
+GNU Libltdl is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with GNU Libltdl; see the file COPYING.LIB.  If not, a
+copy can be downloaded from http://www.gnu.org/licenses/lgpl.html,
+or obtained by writing to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+*/
 
 #if !defined(LT__ARGZ_H)
 #define LT__ARGZ_H 1
@@ -32,7 +41,7 @@
 #  define LT_SCOPE
 #endif
 
-#if defined(_cplusplus)
+#if defined(__cplusplus)
 extern "C" {
 #endif
 
@@ -46,7 +55,7 @@ LT_SCOPE char *	 argz_next	(char *argz,
 				 const char *entry);
 LT_SCOPE void	 argz_stringify	(char *argz, size_t argz_len, int sep);
 
-#if defined(_cplusplus)
+#if defined(__cplusplus)
 }
 #endif
 
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/asinl.c dbg/find6/find/gnulib-cvs/gnulib/lib/asinl.c
