--- dbg/find14/find/AUTHORS	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/AUTHORS	2020-01-28 23:39:42.000000000 -0600
@@ -21,11 +21,12 @@ David J. MacKenzie
 Jim Meyering
 Tim Wood
 Kevin Dalley <kevind@rahul.net>
-Paul Eggert <eggert@twinsun.com>
+Paul Eggert <eggert@cs.ucla.edu>
 James Youngman <jay@gnu.org>
 Jay Plett 
 Paul Sheer
-
+Dmitry V. Levin
+Bas van Gompel
 
 The current maintainer of the findutils package is James Youngman.
 Questions about findutils should be addressed to the mailing list,
diff -up -r dbg/find14/find/autom4te.cache/output.0 dbg/find6/find/autom4te.cache/output.0
