--- /home/jasonhu/Desktop/pydd/patches/expcp/b	2020-03-26 21:58:04.795381315 -0500
+++ /home/jasonhu/Desktop/pydd/patches/expcp2/b	2020-03-26 21:25:27.436601880 -0500
@@ -6,4 +6,9 @@
 123123123
 123123123
 123123123123
-12312312asdfasdfasdf
\ No newline at end of file
+12312312asdfasdfasdf
+
+
+asdfasdf
+a
+asdfasdf
Only in /home/jasonhu/Desktop/pydd/patches/expcp2: bug
diff -u -r /home/jasonhu/Desktop/pydd/patches/expcp/c /home/jasonhu/Desktop/pydd/patches/expcp2/c
