--- dbg/find14/find/gnulib-cvs/gnulib/m4/base64.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/base64.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-# base64.m4 serial 2
-dnl Copyright (C) 2004 Free Software Foundation, Inc.
+# base64.m4 serial 3
+dnl Copyright (C) 2004, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -12,5 +12,5 @@ AC_DEFUN([gl_FUNC_BASE64],
 # Prerequisites of lib/base64.c.
 AC_DEFUN([gl_PREREQ_BASE64], [
   AC_REQUIRE([AC_C_INLINE])
-  AC_REQUIRE([gl_C_RESTRICT])
+  AC_REQUIRE([AC_C_RESTRICT])
 ])
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: bison-i18n.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/bison.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/bison.m4
