--- dbg/find14/find/gnulib-cvs/gnulib/lib/binary-io.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/binary-io.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* Binary mode I/O.
-   Copyright (C) 2001, 2003 Free Software Foundation, Inc.
+   Copyright (C) 2001, 2003, 2005 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,7 +13,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #ifndef _BINARY_H
 #define _BINARY_H
@@ -39,6 +39,7 @@
 #if O_BINARY
 # if !(defined __EMX__ || defined __DJGPP__ || defined __CYGWIN__)
 #  define setmode _setmode
+#  undef fileno
 #  define fileno _fileno
 # endif
 # if defined __DJGPP__ || defined __CYGWIN__
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: byteswap_.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/calloc.c dbg/find6/find/gnulib-cvs/gnulib/lib/calloc.c
