--- dbg/find14/find/gnulib-cvs/gnulib/modules/c-ctype	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/c-ctype	2020-01-28 23:39:42.000000000 -0600
@@ -17,7 +17,7 @@ Include:
 "c-ctype.h"
 
 License:
-GPL
+LGPL
 
 Maintainer:
 Bruno Haible
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: c-ctype-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/chdir-long dbg/find6/find/gnulib-cvs/gnulib/modules/chdir-long
