--- dbg/find14/find/gnulib-cvs/gnulib/m4/c-stack.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/c-stack.m4	2020-01-28 23:39:42.000000000 -0600
@@ -66,7 +66,7 @@ AC_DEFUN([AC_SYS_XSI_STACK_OVERFLOW_HEUR
 	 }
 
 	 static int
-	 c_stack_action (void)
+	 c_stack_action ()
 	 {
 	   stack_t st;
 	   struct sigaction act;
@@ -94,7 +94,7 @@ AC_DEFUN([AC_SYS_XSI_STACK_OVERFLOW_HEUR
 	 }
 
 	 int
-	 main (void)
+	 main ()
 	 {
 	   #if HAVE_SETRLIMIT && defined RLIMIT_STACK
 	   /* Before starting the endless recursion, try to be friendly
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/c-strtod.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/c-strtod.m4
