--- dbg/find14/find/gnulib-cvs/gnulib/modules/c-strtold	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/c-strtold	2020-01-28 23:39:42.000000000 -0600
@@ -9,12 +9,12 @@ m4/c-strtod.m4
 
 Depends-on:
 extensions
+xalloc
 
 configure.ac:
 gl_C_STRTOLD
 
 Makefile.am:
-lib_SOURCES += c-strtod.h c-strtod.c c-strtold.c
 
 Include:
 "c-strtod.h"
@@ -24,4 +24,3 @@ GPL
 
 Maintainer:
 Paul Eggert, Jim Meyering
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/CVS/Entries dbg/find6/find/gnulib-cvs/gnulib/modules/CVS/Entries
