--- dbg/find14/find/gnulib-cvs/gnulib/modules/chdir-long	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/chdir-long	2020-01-28 23:39:42.000000000 -0600
@@ -7,8 +7,11 @@ lib/chdir-long.c
 m4/chdir-long.m4
 
 Depends-on:
+fcntl
 openat
+mempcpy
 memrchr
+stdbool
 
 configure.ac:
 gl_FUNC_CHDIR_LONG
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: chdir-safer
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: check-version
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/chown dbg/find6/find/gnulib-cvs/gnulib/modules/chown
