--- demodir/yesterday/checker.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/checker.py	2020-03-28 19:32:36.003764444 -0500
@@ -471,16 +471,13 @@ class CheckerState(dict):
 
     def get_flipped_state(self):
         new_board = np.copy(self.board)
+        # this is where the bug is, 03/28/2020
         new_board = np.rot90(new_board, 2)
         new_board *= -1
         new_state = CheckerState(new_board, flipped=not self.flipped)
         return new_state
 
-    def first_player_evaluate(self):
-        """
-        First player view evaluation
-        :return:
-        """
+    def evaluate(self):
         if not self.flipped:
             return self.board.sum()
         else:
Only in demodir/today: checker.pyc
Only in demodir/yesterday: csvs
diff -up -r demodir/yesterday/gameInterface.py demodir/today/gameInterface.py
