--- dbg/find14/find/gnulib-cvs/gnulib/m4/cloexec.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/cloexec.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,11 +1,10 @@
-# cloexec.m4 serial 2
-dnl Copyright (C) 2004 Free Software Foundation, Inc.
+#serial 6
+dnl Copyright (C) 2004, 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_CLOEXEC],
 [
-  dnl Prerequisites of lib/cloexec.c.
-  AC_CHECK_HEADERS_ONCE(fcntl.h unistd.h)
+  AC_LIBOBJ([cloexec])
 ])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/closeout.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/closeout.m4
