--- dbg/find14/find/gnulib-cvs/gnulib/modules/cloexec	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/cloexec	2020-01-28 23:39:42.000000000 -0600
@@ -13,7 +13,6 @@ configure.ac:
 gl_CLOEXEC
 
 Makefile.am:
-lib_SOURCES += cloexec.c cloexec.h
 
 Include:
 "cloexec.h"
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/closeout dbg/find6/find/gnulib-cvs/gnulib/modules/closeout
