--- dbg/find14/find/gnulib/lib/closeout.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/closeout.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
-/* Close standard output.
+/* Close standard output and standard error.
 
-   Copyright (C) 1998, 2000, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1998, 2000, 2003, 2004, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -14,7 +14,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #ifndef CLOSEOUT_H
 # define CLOSEOUT_H 1
Binary files dbg/find14/find/gnulib/lib/closeout.o and dbg/find6/find/gnulib/lib/closeout.o differ
Only in dbg/find6/find/gnulib/lib: close-stream.c
Only in dbg/find6/find/gnulib/lib: close-stream.h
Only in dbg/find6/find/gnulib/lib: close-stream.o
Only in dbg/find6/find/gnulib/lib: config.charset
Only in dbg/find6/find/gnulib/lib: configmake.h
Only in dbg/find6/find/gnulib/lib: creat-safer.c
Only in dbg/find6/find/gnulib/lib: creat-safer.o
Only in dbg/find6/find/gnulib/lib: c-strstr.c
Only in dbg/find6/find/gnulib/lib: c-strstr.h
Only in dbg/find6/find/gnulib/lib: c-strstr.o
diff -up -r dbg/find14/find/gnulib/lib/cycle-check.c dbg/find6/find/gnulib/lib/cycle-check.c
