--- dbg/find14/find/gnulib-cvs/gnulib/modules/closeout	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/closeout	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 Description:
-Close stdout, checking for errors.
+Close standard output and standard error, exiting with a diagnostic on error.
 
 Files:
 lib/closeout.h
@@ -7,18 +7,16 @@ lib/closeout.c
 m4/closeout.m4
 
 Depends-on:
-atexit
-gettext
+close-stream
+gettext-h
 error
 quotearg
-fpending
 exitfail
 
 configure.ac:
 gl_CLOSEOUT
 
 Makefile.am:
-lib_SOURCES += closeout.h closeout.c
 
 Include:
 "closeout.h"
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: close-stream
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: config-h
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: configmake
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/copy-file dbg/find6/find/gnulib-cvs/gnulib/modules/copy-file
