--- dbg/find14/find/locate/code.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/locate/code.c	2020-01-28 23:39:42.000000000 -0600
@@ -13,7 +13,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
-   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
    USA.
 */
 
@@ -77,6 +77,7 @@
 #include "locatedb.h"
 #include <getline.h>
 #include "closeout.h"
+#include "gnulib-version.h"
 
 char *xmalloc PARAMS((size_t));
 
@@ -122,8 +123,7 @@ prefix_length (char *s1, char *s2)
 extern char *version_string;
 
 static void
-usage (stream)
-     FILE *stream;
+usage (FILE *stream)
 {
   fprintf (stream, _("\
 Usage: %s [--version | --help]\n\
@@ -164,6 +164,7 @@ main (int argc, char **argv)
   else if (0 == strcmp(argv[1], "--version"))
     {
       printf (_("GNU findutils version %s\n"), version_string);
+      printf (_("Built using GNU gnulib version %s\n"), gnulib_version);
       return 0;
     }
   
Binary files dbg/find14/find/locate/code.o and dbg/find6/find/locate/code.o differ
diff -up -r dbg/find14/find/locate/.deps/bigram.Po dbg/find6/find/locate/.deps/bigram.Po
