--- dbg/find14/find/gnulib-cvs/gnulib/modules/copy-file	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/copy-file	2020-01-28 23:39:42.000000000 -0600
@@ -10,9 +10,11 @@ Depends-on:
 error
 safe-read
 full-write
+acl
 binary-io
 exit
-gettext
+gettext-h
+unistd
 
 configure.ac:
 gl_COPY_FILE
@@ -28,4 +30,3 @@ GPL
 
 Maintainer:
 Bruno Haible
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: COPYING
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: crc
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: crc-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: crypto
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: csharpcomp
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: csharpcomp-script
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: csharpexec
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: csharpexec-script
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/c-stack dbg/find6/find/gnulib-cvs/gnulib/modules/c-stack
