--- dbg/find14/find/gnulib-cvs/gnulib/modules/cycle-check	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/cycle-check	2020-01-28 23:39:42.000000000 -0600
@@ -4,14 +4,18 @@ help detect directory cycles efficiently
 Files:
 lib/cycle-check.c
 lib/cycle-check.h
-lib/dev-ino.h
+m4/cycle-check.m4
 
 Depends-on:
+dev-ino
+same-inode
+stdbool
+stdint
 
 configure.ac:
+gl_CYCLE_CHECK
 
 Makefile.am:
-lib_SOURCES += cycle-check.c cycle-check.h dev-ino.h
 
 Include:
 "cycle-check.h"
@@ -21,4 +25,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: dev-ino
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: d-ino
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/dirfd dbg/find6/find/gnulib-cvs/gnulib/modules/dirfd
