--- dbg/find14/diff.sh	2020-01-28 23:48:57.000000000 -0600
+++ dbg/find6/diff.sh	2020-01-28 23:40:10.000000000 -0600
@@ -1,9 +1,9 @@
 #!/bin/bash
-cd /root/Desktop/find14/find/
-#git diff HEAD > ../find14.patch
-#diff -x "*~" -Naur  /root/Desktop/find14/find > ../find14.patch
-git diff find14 > ../find14.patch
+cd /root/Desktop/find6/find/
+#git diff HEAD > ../find6.patch
+#diff -x "*~" -Naur  /root/Desktop/find6/find > ../find6.patch
+git diff find6 > ../find6.patch
 echo ""
-echo "Please copy the contents of find14.patch into"
-echo "the appropriate field at http://bit.do/find-ff248a20."
+echo "Please copy the contents of find6.patch into"
+echo "the appropriate field at http://bit.do/find-091557f6."
 echo ""
Only in dbg/find14: exp
diff -up -r dbg/find14/find/ABOUT-NLS dbg/find6/find/ABOUT-NLS
