--- dbg/find14/find/gnulib-cvs/gnulib/lib/dup2.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/dup2.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,6 @@
 /* Duplicate an open file descriptor to a specified file descriptor.
-   Copyright (C) 1999, 2004 Free Software Foundation, Inc.
+
+   Copyright (C) 1999, 2004, 2005, 2006, 2007 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,23 +14,17 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* written by Paul Eggert */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
-#include <errno.h>
+/* Specification.  */
+#include <unistd.h>
 
-#if HAVE_FCNTL_H
-# include <fcntl.h>
-#endif
-
-#if HAVE_UNISTD_H
-# include <unistd.h>
-#endif
+#include <errno.h>
+#include <fcntl.h>
 
 #ifndef F_DUPFD
 static int
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/dup-safer.c dbg/find6/find/gnulib-cvs/gnulib/lib/dup-safer.c
