--- dbg/find14/find/gnulib-cvs/gnulib/modules/euidaccess	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/euidaccess	2020-01-28 23:39:42.000000000 -0600
@@ -4,24 +4,25 @@ euidaccess() function: check effective u
 Files:
 lib/euidaccess.h
 lib/euidaccess.c
-lib/stat-macros.h
 m4/euidaccess.m4
 
 Depends-on:
 group-member
+sys_stat
 
 configure.ac:
 gl_FUNC_EUIDACCESS
 
 Makefile.am:
-lib_SOURCES += euidaccess.h stat-macros.h
 
 Include:
 "euidaccess.h"
 
+Link:
+$(LIB_EACCESS)
+
 License:
 LGPL
 
 Maintainer:
 Jim Meyering, glibc
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/exclude dbg/find6/find/gnulib-cvs/gnulib/modules/exclude
