--- dbg/find14/find/find/testsuite/find.gnu/execdir-root-only.exp	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/find/testsuite/find.gnu/execdir-root-only.exp	2020-01-28 23:39:42.000000000 -0600
@@ -16,4 +16,6 @@
 
 
 # This test checks the second of these two cases.
-find_start p { / -mindepth 0 -maxdepth 0 -execdir echo == "{}" + }
+if { [ safe_path ] } {
+    find_start p { / -mindepth 0 -maxdepth 0 -execdir echo == "{}" + }
+}
Only in dbg/find6/find/find/testsuite/find.gnu: false.exp
Only in dbg/find6/find/find/testsuite/find.gnu: false.xo
Only in dbg/find6/find/find/testsuite/find.gnu: follow-basic.exp
Only in dbg/find6/find/find/testsuite/find.gnu: follow-basic.xo
Only in dbg/find6/find/find/testsuite/find.gnu: gnuand.exp
Only in dbg/find6/find/find/testsuite/find.gnu: gnuand.xo
Only in dbg/find6/find/find/testsuite/find.gnu: gnunot.exp
Only in dbg/find6/find/find/testsuite/find.gnu: gnunot.xo
Only in dbg/find6/find/find/testsuite/find.gnu: gnu-or.exp
Only in dbg/find6/find/find/testsuite/find.gnu: gnu-or.xo
Only in dbg/find6/find/find/testsuite/find.gnu: ilname.exp
Only in dbg/find6/find/find/testsuite/find.gnu: ilname.xo
Only in dbg/find6/find/find/testsuite/find.gnu: iname.exp
Only in dbg/find6/find/find/testsuite/find.gnu: iname.xo
Only in dbg/find6/find/find/testsuite/find.gnu: inum.exp
Only in dbg/find6/find/find/testsuite/find.gnu: inum.xo
Only in dbg/find6/find/find/testsuite/find.gnu: ipath.exp
Only in dbg/find6/find/find/testsuite/find.gnu: ipath.xo
Only in dbg/find6/find/find/testsuite/find.gnu: iregex1.exp
Only in dbg/find6/find/find/testsuite/find.gnu: iregex1.xo
Only in dbg/find6/find/find/testsuite/find.gnu: iwholename.exp
Only in dbg/find6/find/find/testsuite/find.gnu: iwholename.xo
Only in dbg/find6/find/find/testsuite/find.gnu: lname.exp
Only in dbg/find6/find/find/testsuite/find.gnu: lname.xo
Only in dbg/find6/find/find/testsuite/find.gnu: path.exp
Only in dbg/find6/find/find/testsuite/find.gnu: path.xo
Only in dbg/find6/find/find/testsuite/find.gnu: perm000.exp
Only in dbg/find6/find/find/testsuite/find.gnu: perm000.xo
Only in dbg/find6/find/find/testsuite/find.gnu: perm-slash.exp
Only in dbg/find6/find/find/testsuite/find.gnu: perm-slash.xo
Only in dbg/find6/find/find/testsuite/find.gnu: print0.exp
Only in dbg/find6/find/find/testsuite/find.gnu: print0.xo
Only in dbg/find6/find/find/testsuite/find.gnu: printfHdfl.exp
Only in dbg/find6/find/find/testsuite/find.gnu: printfHdfl.xo
Only in dbg/find6/find/find/testsuite/find.gnu: printf-h.exp
Only in dbg/find6/find/find/testsuite/find.gnu: printf-h.xo
diff -up -r dbg/find14/find/find/testsuite/find.gnu/printf-symlink.exp dbg/find6/find/find/testsuite/find.gnu/printf-symlink.exp
