--- dbg/find14/find/gnulib-cvs/gnulib/modules/execute	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/execute	2020-01-28 23:39:42.000000000 -0600
@@ -12,9 +12,10 @@ error
 exit
 fatal-signal
 wait-process
-gettext
+gettext-h
 stdbool
 strpbrk
+unistd
 
 configure.ac:
 gl_EXECUTE
@@ -30,4 +31,3 @@ GPL
 
 Maintainer:
 Bruno Haible
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/exit dbg/find6/find/gnulib-cvs/gnulib/modules/exit
