--- dbg/find14/find/gnulib-cvs/gnulib/modules/exit	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/exit	2020-01-28 23:39:42.000000000 -0600
@@ -2,17 +2,16 @@ Description:
 exit() function: program termination.
 
 Files:
-lib/exit.h
 
 Depends-on:
+stdlib
 
 configure.ac:
 
 Makefile.am:
-lib_SOURCES += exit.h
 
 Include:
-"exit.h"
+<stdlib.h>
 
 License:
 GPL
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/exitfail dbg/find6/find/gnulib-cvs/gnulib/modules/exitfail
