--- dbg/find14/find/gnulib-cvs/gnulib/modules/exitfail	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/exitfail	2020-01-28 23:39:42.000000000 -0600
@@ -13,7 +13,6 @@ configure.ac:
 gl_EXITFAIL
 
 Makefile.am:
-lib_SOURCES += exitfail.h exitfail.c
 
 Include:
 "exitfail.h"
@@ -23,4 +22,3 @@ GPL
 
 Maintainer:
 Paul Eggert
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/extensions dbg/find6/find/gnulib-cvs/gnulib/modules/extensions
