--- dbg/find14/find/gnulib-cvs/gnulib/modules/extensions	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/extensions	2020-01-28 23:39:42.000000000 -0600
@@ -6,16 +6,17 @@ m4/extensions.m4
 
 Depends-on:
 
+configure.ac-early:
+AC_REQUIRE([gl_USE_SYSTEM_EXTENSIONS])
+
 configure.ac:
-dnl gl_USE_SYSTEM_EXTENSIONS must be added quite early to configure.ac.
 
 Makefile.am:
 
 Include:
 
 License:
-GPL
+LGPL
 
 Maintainer:
 Paul Eggert and Jim Meyering
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/fatal-signal dbg/find6/find/gnulib-cvs/gnulib/modules/fatal-signal
