--- dbg/find14/find/gnulib-cvs/gnulib/m4/fatal-signal.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/fatal-signal.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,13 +1,13 @@
-# fatal-signal.m4 serial 3
-dnl Copyright (C) 2003-2004 Free Software Foundation, Inc.
+# fatal-signal.m4 serial 4
+dnl Copyright (C) 2003-2004, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_FATAL_SIGNAL],
 [
-  AC_REQUIRE([gt_SIGNALBLOCKING])
   AC_REQUIRE([gt_TYPE_SIG_ATOMIC_T])
   AC_CHECK_HEADERS_ONCE(unistd.h)
-  AC_CHECK_FUNCS(raise sigaction)
+  AC_CHECK_FUNCS_ONCE(raise)
+  AC_CHECK_FUNCS(sigaction)
 ])
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: fchdir.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: fcntl_h.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/fcntl-safer.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/fcntl-safer.m4
