--- dbg/find14/find/gnulib-cvs/gnulib/lib/fchown-stub.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/fchown-stub.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,4 +1,5 @@
 #include <config.h>
+
 #include <sys/types.h>
 #include <errno.h>
 
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: fcntl_.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: fcntl--.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/fcntl-safer.h dbg/find6/find/gnulib-cvs/gnulib/lib/fcntl-safer.h
