--- dbg/find14/find/gnulib-cvs/gnulib/modules/fcntl-safer	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/fcntl-safer	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,9 @@
 Description:
-File control functions that avoid clobbering std{in,out,err}.
+File descriptor functions that avoid clobbering STD{IN,OUT,ERR}_FILENO.
 
 Files:
+lib/fcntl--.h
+lib/creat-safer.c
 lib/fcntl-safer.h
 lib/open-safer.c
 m4/fcntl-safer.m4
@@ -11,9 +13,9 @@ unistd-safer
 
 configure.ac:
 gl_FCNTL_SAFER
+gl_MODULE_INDICATOR([fcntl-safer])
 
 Makefile.am:
-lib_SOURCES += fcntl-safer.h open-safer.c
 
 Include:
 "fcntl-safer.h"
@@ -23,4 +25,3 @@ GPL
 
 Maintainer:
 Paul Eggert, Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fcntl-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fdl
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/filemode dbg/find6/find/gnulib-cvs/gnulib/modules/filemode
