--- dbg/find14/find/gnulib/lib/filemode.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/filemode.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
 /* Make a string describing file modes.
 
-   Copyright (C) 1998, 1999, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2003, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -14,12 +14,20 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #ifndef FILEMODE_H_
 
 # include <sys/types.h>
+# include <sys/stat.h>
 
-void mode_string (mode_t mode, char *str);
+# if HAVE_DECL_STRMODE
+#  include <string.h> /* FreeBSD, OpenBSD */
+#  include <unistd.h> /* NetBSD */
+# else
+void strmode (mode_t mode, char *str);
+# endif
+
+void filemodestring (struct stat const *statp, char *str);
 
 #endif
Binary files dbg/find14/find/gnulib/lib/filemode.o and dbg/find6/find/gnulib/lib/filemode.o differ
Only in dbg/find6/find/gnulib/lib: filenamecat.c
Only in dbg/find6/find/gnulib/lib: filenamecat.h
Only in dbg/find6/find/gnulib/lib: filenamecat.o
diff -up -r dbg/find14/find/gnulib/lib/fnmatch.c dbg/find6/find/gnulib/lib/fnmatch.c
