--- dbg/find14/find/gnulib-cvs/gnulib/modules/filemode	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/filemode	2020-01-28 23:39:42.000000000 -0600
@@ -5,16 +5,16 @@ file.
 Files:
 lib/filemode.h
 lib/filemode.c
-lib/stat-macros.h
 m4/filemode.m4
+m4/st_dm_mode.m4
 
 Depends-on:
+sys_stat
 
 configure.ac:
 gl_FILEMODE
 
 Makefile.am:
-lib_SOURCES += filemode.h filemode.c stat-macros.h
 
 Include:
 "filemode.h"
@@ -24,4 +24,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: filename
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: filenamecat
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/file-type dbg/find6/find/gnulib-cvs/gnulib/modules/file-type
