--- dbg/find14/find/gnulib-cvs/gnulib/modules/findprog	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/findprog	2020-01-28 23:39:42.000000000 -0600
@@ -10,7 +10,8 @@ m4/eaccess.m4
 Depends-on:
 stdbool
 xalloc
-pathname
+filename
+unistd
 
 configure.ac:
 gl_FINDPROG
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: flexmember
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/fnmatch dbg/find6/find/gnulib-cvs/gnulib/modules/fnmatch
