--- dbg/find14/find/gnulib-cvs/gnulib/lib/fopen-safer.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/fopen-safer.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,6 @@
 /* Invoke fopen, but avoid some glitches.
-   Copyright (C) 2001, 2004 Free Software Foundation, Inc.
+
+   Copyright (C) 2001, 2004, 2005, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,27 +14,22 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by Paul Eggert.  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
-#if HAVE_UNISTD_H
-# include <unistd.h>
-#endif
-#include <unistd-safer.h>
+#include "stdio-safer.h"
+
+#include <errno.h>
+#include <unistd.h>
+#include "unistd-safer.h"
 
 #ifndef STDERR_FILENO
 # define STDERR_FILENO 2
 #endif
 
-#include <errno.h>
-#include <stdio.h>
-#include <stdio-safer.h>
-
 /* Like fopen, but do not return stdin, stdout, or stderr.  */
 
 FILE *
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/__fpending.c dbg/find6/find/gnulib-cvs/gnulib/lib/__fpending.c
