--- dbg/find14/find/gnulib/m4/fpending.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/m4/fpending.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
-#serial 5
+#serial 10
 
-# Copyright (C) 2000, 2001, 2004 Free Software Foundation, Inc.
+# Copyright (C) 2000, 2001, 2004, 2005, 2006 Free Software Foundation, Inc.
 # This file is free software; the Free Software Foundation
 # gives unlimited permission to copy and/or distribute it,
 # with or without modifications, as long as this notice is preserved.
@@ -15,9 +15,10 @@ dnl we have to grub around in the FILE s
 
 AC_DEFUN([gl_FUNC_FPENDING],
 [
-  AC_CHECK_HEADERS(stdio_ext.h)
+  AC_CHECK_HEADERS_ONCE(stdio_ext.h)
   AC_REPLACE_FUNCS([__fpending])
   fp_headers='
+#     include <stdio.h>
 #     if HAVE_STDIO_EXT_H
 #      include <stdio_ext.h>
 #     endif
@@ -40,7 +41,7 @@ AC_DEFUN([gl_FUNC_FPENDING],
 	    'fp->_p - fp->_bf._base'				\
 								\
 	    '# SCO, Unixware'					\
-	    'fp->__ptr - fp->__base'				\
+	    '(fp->__ptr ? fp->__ptr - fp->__base : 0)'		\
 								\
 	    '# old glibc?'					\
 	    'fp->__bufp - fp->__buffer'				\
diff -up -r dbg/find14/find/gnulib/m4/fstypename.m4 dbg/find6/find/gnulib/m4/fstypename.m4
