--- dbg/find14/find/gnulib-cvs/gnulib/modules/fpending	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/fpending	2020-01-28 23:39:42.000000000 -0600
@@ -12,7 +12,6 @@ configure.ac:
 gl_FUNC_FPENDING
 
 Makefile.am:
-lib_SOURCES += __fpending.h
 
 Include:
 "__fpending.h"
@@ -22,4 +21,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fpieee
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fprintf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fprintf-posix-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fprintftime
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fpucw
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: frexp
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: frexpl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: frexpl-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: frexp-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fseterr
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: fseterr-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/fsusage dbg/find6/find/gnulib-cvs/gnulib/modules/fsusage
