--- dbg/find14/find/gnulib-cvs/gnulib/m4/free.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/free.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
 # Check whether free (NULL) is supposed to work.
 
-# Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+# Copyright (C) 2003, 2004, 2005 Free Software Foundation, Inc.
 # This file is free software; the Free Software Foundation
 # gives unlimited permission to copy and/or distribute it,
 # with or without modifications, as long as this notice is preserved.
@@ -16,14 +16,11 @@
 
 AC_DEFUN([gl_FUNC_FREE],
 [
-  AC_CHECK_HEADERS_ONCE(unistd.h)
   AC_CACHE_CHECK([whether free (NULL) is known to work],
     [gl_cv_func_free],
     [AC_COMPILE_IFELSE(
        [AC_LANG_PROGRAM(
-	  [[@%:@if HAVE_UNISTD_H
-	      @%:@include <unistd.h>
-	    @%:@endif]],
+	  [[@%:@include <unistd.h>]],
 	  [[@%:@if _POSIX_VERSION < 199009L && \
 	        (defined unix || defined _unix || defined _unix_ \
 	         || defined __unix || defined __unix__)
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: frexpl.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: frexp.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/fstypename.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/fstypename.m4
