--- dbg/find14/find/gnulib-cvs/gnulib/lib/ftruncate.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/ftruncate.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,9 +1,10 @@
 /* ftruncate emulations that work on some System V's.
    This file is in the public domain.  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
+
+/* Specification.  */
+#include <unistd.h>
 
 #include <sys/types.h>
 #include <fcntl.h>
@@ -23,9 +24,6 @@ ftruncate (int fd, off_t length)
 
 #  include <sys/stat.h>
 #  include <errno.h>
-#  if HAVE_UNISTD_H
-#   include <unistd.h>
-#  endif
 
 int
 ftruncate (int fd, off_t length)
@@ -68,7 +66,7 @@ ftruncate (int fd, off_t length)
 }
 
 # else /* not F_CHSIZE nor F_FREESP */
-#  if HAVE_CHSIZE
+#  if HAVE_CHSIZE                      /* native Windows, e.g. mingw */
 
 int
 ftruncate (int fd, off_t length)
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: fts.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: fts-cycle.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: fts_.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/full-read.c dbg/find6/find/gnulib-cvs/gnulib/lib/full-read.c
