--- dbg/find14/find/gnulib-cvs/gnulib/lib/full-write.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/full-write.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,7 @@
 /* An interface to read and write that retries (if necessary) until complete.
 
    Copyright (C) 1993, 1994, 1997, 1998, 1999, 2000, 2001, 2002, 2003,
-   2004 Free Software Foundation, Inc.
+   2004, 2005, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -15,11 +15,9 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 /* Specification.  */
 #ifdef FULL_READ
@@ -62,7 +60,7 @@ size_t
 full_rw (int fd, const void *buf, size_t count)
 {
   size_t total = 0;
-  const char *ptr = buf;
+  const char *ptr = (const char *) buf;
 
   while (count > 0)
     {
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/full-write.h dbg/find6/find/gnulib-cvs/gnulib/lib/full-write.h
