--- dbg/find14/find/gnulib-cvs/gnulib/modules/fwriteerror	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/fwriteerror	2020-01-28 23:39:42.000000000 -0600
@@ -6,8 +6,10 @@ lib/fwriteerror.h
 lib/fwriteerror.c
 
 Depends-on:
+stdbool
 
 configure.ac:
+gl_MODULE_INDICATOR([fwriteerror])
 
 Makefile.am:
 lib_SOURCES += fwriteerror.h fwriteerror.c
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: gc
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: gendocs
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getaddrinfo dbg/find6/find/gnulib-cvs/gnulib/modules/getaddrinfo
