--- demodir/yesterday/gameInterface.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/gameInterface.py	2019-11-29 23:50:49.000000000 -0600
@@ -23,3 +23,5 @@ def machine_versus_machine():
     ab.start_game()
 
 
+if __name__ == '__main__':
+    machine_versus_machine()
\ No newline at end of file
Only in demodir/yesterday: LICENSE
Only in demodir/yesterday: logplotter.R
Only in demodir/yesterday: logprocessor.py
diff -up -r demodir/yesterday/mcts.py demodir/today/mcts.py
