--- dbg/find14/find/gnulib-cvs/gnulib/modules/getcwd	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getcwd	2020-01-28 23:39:42.000000000 -0600
@@ -2,23 +2,27 @@ Description:
 Return the current working directory.
 
 Files:
-lib/getcwd.h
 lib/getcwd.c
-m4/d-ino.m4
+m4/getcwd-abort-bug.m4
 m4/getcwd-path-max.m4
 m4/getcwd.m4
 
 Depends-on:
 mempcpy
+d-ino
+dirfd
+extensions
+stdbool
+unistd
 
 configure.ac:
 gl_FUNC_GETCWD
+gl_UNISTD_MODULE_INDICATOR([getcwd])
 
 Makefile.am:
-lib_SOURCES += getcwd.h
 
 Include:
-"getcwd.h"
+<unistd.h>
 
 License:
 GPL
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getdate dbg/find6/find/gnulib-cvs/gnulib/modules/getdate
