--- dbg/find14/find/gnulib-cvs/gnulib/modules/getdate	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getdate	2020-01-28 23:39:42.000000000 -0600
@@ -10,20 +10,21 @@ m4/tm_gmtoff.m4
 m4/getdate.m4
 
 Depends-on:
-timespec
 stdbool
 gettime
 mktime
 setenv
+time
 xalloc
 
 configure.ac:
 gl_GETDATE
 
 Makefile.am:
-lib_SOURCES += getdate.h getdate.y
+lib_SOURCES += getdate.y
 BUILT_SOURCES += getdate.c
 MAINTAINERCLEANFILES += getdate.c
+EXTRA_DIST += getdate.c
 
 Include:
 "getdate.h"
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: getdelim
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getdomainname dbg/find6/find/gnulib-cvs/gnulib/modules/getdomainname
