--- dbg/find14/find/gnulib-cvs/gnulib/lib/getdate.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/getdate.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,7 @@
 /* Parse a string into an internal time stamp.
 
-   Copyright (C) 1995, 1997, 1998, 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1995, 1997, 1998, 2003, 2004, 2007 Free Software
+   Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -14,13 +15,9 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
-
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #include <stdbool.h>
-#include "timespec.h"
+#include <time.h>
 
 bool get_date (struct timespec *, char const *, struct timespec const *);
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/getdate.y dbg/find6/find/gnulib-cvs/gnulib/lib/getdate.y
