--- dbg/find14/find/gnulib-cvs/gnulib/lib/getgroups.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/getgroups.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,6 @@
 /* provide consistent interface to getgroups for systems that don't allow N==0
-   Copyright (C) 1996, 1999, 2003 Free Software Foundation, Inc.
+
+   Copyright (C) 1996, 1999, 2003, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,11 +14,12 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* written by Jim Meyering */
 
 #include <config.h>
+
 #include <stdio.h>
 #include <sys/types.h>
 #include <errno.h>
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/gethostname.c dbg/find6/find/gnulib-cvs/gnulib/lib/gethostname.c
