--- dbg/find14/find/gnulib-cvs/gnulib/modules/gethostname	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/gethostname	2020-01-28 23:39:42.000000000 -0600
@@ -15,8 +15,7 @@ Makefile.am:
 Include:
 
 License:
-GPL
+LGPL
 
 Maintainer:
 all
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: gethrxtime
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getline dbg/find6/find/gnulib-cvs/gnulib/modules/getline
