--- dbg/find14/find/gnulib-cvs/gnulib/modules/getline	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getline	2020-01-28 23:39:42.000000000 -0600
@@ -4,27 +4,21 @@ Read a line from a stream.
 Files:
 lib/getline.h
 lib/getline.c
-lib/getndelim2.h
-lib/getndelim2.c
 m4/getline.m4
-m4/getndelim2.m4
-m4/ssize_t.m4
 
 Depends-on:
+getdelim
 
 configure.ac:
-AM_FUNC_GETLINE
+gl_FUNC_GETLINE
 
 Makefile.am:
-lib_SOURCES += getline.h
-EXTRA_DIST += getndelim2.h getndelim2.c
 
 Include:
 "getline.h"
 
 License:
-GPL
+LGPL
 
 Maintainer:
-all
-
+Simon Josefsson
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getloadavg dbg/find6/find/gnulib-cvs/gnulib/modules/getloadavg
