--- dbg/find14/find/gnulib-cvs/gnulib/modules/getnline	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getnline	2020-01-28 23:39:42.000000000 -0600
@@ -5,10 +5,10 @@ Files:
 lib/getnline.h
 lib/getnline.c
 m4/getnline.m4
-m4/ssize_t.m4
 
 Depends-on:
 getndelim2
+ssize_t
 
 configure.ac:
 gl_GETNLINE
@@ -24,4 +24,3 @@ GPL
 
 Maintainer:
 all
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getopt dbg/find6/find/gnulib-cvs/gnulib/modules/getopt
