--- dbg/find14/find/gnulib-cvs/gnulib/lib/getndelim2.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/getndelim2.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,7 @@
 /* getndelim2 - Read a line from a stream, stopping at one of 2 delimiters,
    with bounded memory allocation.
 
-   Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+   Copyright (C) 2003, 2004, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -15,7 +15,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #ifndef GETNDELIM2_H
 #define GETNDELIM2_H 1
@@ -31,8 +31,8 @@
    the buffer as necessary, but if NMAX is not GETNLINE_NO_LIMIT
    then do not allocate more than NMAX bytes; if the line is longer
    than that, read and discard the extra bytes.  Stop reading after
-   after the first occurrence of DELIM1 or DELIM2, whichever comes
-   first; a delimiter equal to EOF stands for no delimiter.  Read the
+   the first occurrence of DELIM1 or DELIM2, whichever comes first;
+   a delimiter equal to EOF stands for no delimiter.  Read the
    input bytes from STREAM.
    Return the number of bytes read and stored at *LINEPTR + OFFSET (not
    including the NUL terminator), or -1 on error or EOF.  */
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/getnline.c dbg/find6/find/gnulib-cvs/gnulib/lib/getnline.c
