--- dbg/find14/find/gnulib-cvs/gnulib/modules/getopt	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getopt	2020-01-28 23:39:42.000000000 -0600
@@ -9,21 +9,22 @@ lib/getopt_int.h
 m4/getopt.m4
 
 Depends-on:
-gettext
+gettext-h
+unistd
 
 configure.ac:
 gl_GETOPT
 
 Makefile.am:
 BUILT_SOURCES += $(GETOPT_H)
-EXTRA_DIST += getopt_.h getopt_int.h
 
-# We need the following in order to create an <getopt.h> when the system
+# We need the following in order to create <getopt.h> when the system
 # doesn't have one that works with the given compiler.
-all-local $(lib_OBJECTS): $(GETOPT_H)
 getopt.h: getopt_.h
-	cp $(srcdir)/getopt_.h $@-t
-	mv $@-t $@
+	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
+	  cat $(srcdir)/getopt_.h; \
+	} > $@-t
+	mv -f $@-t $@
 MOSTLYCLEANFILES += getopt.h getopt.h-t
 
 Include:
@@ -34,4 +35,3 @@ LGPL
 
 Maintainer:
 all, glibc
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getpagesize dbg/find6/find/gnulib-cvs/gnulib/modules/getpagesize
