--- dbg/find14/find/gnulib-cvs/gnulib/modules/getpass	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getpass	2020-01-28 23:39:42.000000000 -0600
@@ -14,7 +14,6 @@ configure.ac:
 gl_FUNC_GETPASS
 
 Makefile.am:
-lib_SOURCES += getpass.h
 
 Include:
 "getpass.h"
@@ -24,4 +23,3 @@ LGPL
 
 Maintainer:
 Jim Meyering, glibc
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getpass-gnu dbg/find6/find/gnulib-cvs/gnulib/modules/getpass-gnu
