--- dbg/find14/find/gnulib-cvs/gnulib/lib/getsubopt.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/getsubopt.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-/* Parse comma separate list into words.
-   Copyright (C) 1996, 1997, 1999, 2004 Free Software Foundation, Inc.
+/* Parse comma separated list into words.
+   Copyright (C) 1996, 1997, 1999, 2004, 2007 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Ulrich Drepper <drepper@cygnus.com>, 1996.
 
@@ -15,7 +15,11 @@
 
    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
+
+#if !_LIBC
+# include <config.h>
+#endif
 
 #include <stdlib.h>
 #include <string.h>
@@ -24,7 +28,6 @@
 /* This code is written for inclusion in gnu-libc, and uses names in
    the namespace reserved for libc.  If we're compiling in gnulib,
    define those names to be the normal ones instead.  */
-# include "strchrnul.h"
 # undef __strchrnul
 # define __strchrnul strchrnul
 #endif
Only in dbg/find14/find/gnulib-cvs/gnulib/lib: getsubopt.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/gettext.h dbg/find6/find/gnulib-cvs/gnulib/lib/gettext.h
