--- dbg/find14/find/gnulib-cvs/gnulib/modules/getsubopt	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getsubopt	2020-01-28 23:39:42.000000000 -0600
@@ -1,22 +1,22 @@
 Description:
-getsubopt: Parse comma separate list into words.
+getsubopt: Parse comma separated list into words.
 
 Files:
-lib/getsubopt.h
 lib/getsubopt.c
 m4/getsubopt.m4
 
 Depends-on:
 strchrnul
+stdlib
 
 configure.ac:
 gl_FUNC_GETSUBOPT
+gl_STDLIB_MODULE_INDICATOR([getsubopt])
 
 Makefile.am:
-lib_SOURCES += getsubopt.h
 
 Include:
-"getsubopt.h"
+<stdlib.h>
 
 License:
 LGPL
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/gettext dbg/find6/find/gnulib-cvs/gnulib/modules/gettext
