--- dbg/find14/find/gnulib-cvs/gnulib/modules/gettimeofday	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/gettimeofday	2020-01-28 23:39:42.000000000 -0600
@@ -6,9 +6,10 @@ lib/gettimeofday.c
 m4/gettimeofday.m4
 
 Depends-on:
+sys_time
 
 configure.ac:
-AC_FUNC_GETTIMEOFDAY_CLOBBER
+gl_FUNC_GETTIMEOFDAY
 
 Makefile.am:
 
@@ -16,8 +17,7 @@ Include:
 <sys/time.h>
 
 License:
-GPL
+LGPL
 
 Maintainer:
-Jim Meyering
-
+Paul Eggert, Jim Meyering
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: gettimeofday-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getugroups dbg/find6/find/gnulib-cvs/gnulib/modules/getugroups
