--- dbg/find14/find/gnulib-cvs/gnulib/lib/getugroups.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/getugroups.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,7 @@
 /* getugroups.c -- return a list of the groups a user is in
 
-   Copyright (C) 1990, 1991, 1998, 1999, 2000, 2003, 2004 Free
-   Software Foundation.
+   Copyright (C) 1990, 1991, 1998, 1999, 2000, 2003, 2004, 2005, 2006
+   Free Software Foundation.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -15,29 +15,24 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by David MacKenzie. */
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include <sys/types.h>
 #include <stdio.h> /* grp.h on alpha OSF1 V2.0 uses "FILE *". */
 #include <grp.h>
 
-#ifdef HAVE_UNISTD_H
-# include <unistd.h>
-#endif
+#include <unistd.h>
 
 #include <errno.h>
 #ifndef EOVERFLOW
 # define EOVERFLOW EINVAL
 #endif
 
-/* setgrent, getgrent, and endgrent are not specified by POSIX.1,
-   so header files might not declare them.
+/* Some old header files might not declare setgrent, getgrent, and endgrent.
    If you don't have them at all, we can't implement this function.
    You lose!  */
 struct group *getgrent ();
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/getusershell.c dbg/find6/find/gnulib-cvs/gnulib/lib/getusershell.c
