--- dbg/find14/find/gnulib-cvs/gnulib/modules/getusershell	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getusershell	2020-01-28 23:39:42.000000000 -0600
@@ -6,10 +6,11 @@ lib/getusershell.c
 m4/getusershell.m4
 
 Depends-on:
+fopen-safer
 xalloc
 
 configure.ac:
-gl_PREREQ_GETUSERSHELL
+gl_FUNC_GETUSERSHELL
 
 Makefile.am:
 
@@ -20,4 +21,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: glob
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: gnupload
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: gpl
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/group-member dbg/find6/find/gnulib-cvs/gnulib/modules/group-member
