--- dbg/find14/find/gnulib-cvs/gnulib/modules/getugroups	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/getugroups	2020-01-28 23:39:42.000000000 -0600
@@ -11,7 +11,6 @@ configure.ac:
 gl_GETUGROUPS
 
 Makefile.am:
-lib_SOURCES += getugroups.c
 
 Include:
 
@@ -20,4 +19,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/getusershell dbg/find6/find/gnulib-cvs/gnulib/modules/getusershell
