--- dbg/find14/find/gnulib-cvs/gnulib/lib/hard-locale.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/hard-locale.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,7 @@
 /* hard-locale.c -- Determine whether a locale is hard.
 
-   Copyright (C) 1997, 1998, 1999, 2002, 2003, 2004 Free Software
-   Foundation, Inc.
+   Copyright (C) 1997, 1998, 1999, 2002, 2003, 2004, 2006, 2007 Free
+   Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -15,11 +15,9 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include "hard-locale.h"
 
@@ -27,8 +25,6 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include "strdup.h"
-
 #ifdef __GLIBC__
 # define GLIBC_VERSION __GLIBC__
 #else
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/hard-locale.h dbg/find6/find/gnulib-cvs/gnulib/lib/hard-locale.h
