--- dbg/find14/find/gnulib-cvs/gnulib/lib/hash.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/hash.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
 /* hash - hashing table processing.
 
-   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003, 2004 Free
+   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2006 Free
    Software Foundation, Inc.
 
    Written by Jim Meyering, 1992.
@@ -17,16 +17,14 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* A generic hash table package.  */
 
 /* Define USE_OBSTACK to 1 if you want the allocator to use obstacks instead
    of malloc.  If you change USE_OBSTACK, you have to recompile!  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include "hash.h"
 #include "xalloc.h"
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/hash.h dbg/find6/find/gnulib-cvs/gnulib/lib/hash.h
