--- dbg/find14/find/gnulib-cvs/gnulib/modules/iconv	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/iconv	2020-01-28 23:39:42.000000000 -0600
@@ -3,12 +3,9 @@ Character set conversion.
 
 Files:
 m4/iconv.m4
-m4/lib-ld.m4
-m4/lib-link.m4
-m4/lib-prefix.m4
-config/config.rpath
 
 Depends-on:
+havelib
 
 configure.ac:
 AM_ICONV
@@ -16,7 +13,12 @@ AM_ICONV
 Makefile.am:
 
 Include:
-<iconv.h>
+#if HAVE_ICONV
+# include <iconv.h>
+#endif
+
+Link:
+$(LTLIBICONV) when linking with libtool, $(LIBICONV) otherwise
 
 License:
 LGPL
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: iconv_open
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: iconv-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/idcache dbg/find6/find/gnulib-cvs/gnulib/modules/idcache
