--- dbg/find14/find/gnulib-cvs/gnulib/modules/idcache	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/idcache	2020-01-28 23:39:42.000000000 -0600
@@ -6,13 +6,13 @@ lib/idcache.c
 m4/idcache.m4
 
 Depends-on:
+flexmember
 xalloc
 
 configure.ac:
 gl_IDCACHE
 
 Makefile.am:
-lib_SOURCES += idcache.c
 
 Include:
 
@@ -21,4 +21,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: imaxabs
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: imaxdiv
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: inet_ntop
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: inet_pton
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: inline
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: intprops
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/inttostr dbg/find6/find/gnulib-cvs/gnulib/modules/inttostr
