--- dbg/find14/find/gnulib-cvs/gnulib/m4/inttostr.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/inttostr.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,21 +1,25 @@
-# inttostr.m4 serial 3
-dnl Copyright (C) 2004 Free Software Foundation, Inc.
+#serial 7
+dnl Copyright (C) 2004, 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_INTTOSTR],
 [
+  AC_LIBOBJ([imaxtostr])
+  AC_LIBOBJ([offtostr])
+  AC_LIBOBJ([umaxtostr])
+  AC_LIBOBJ([uinttostr])
+
   gl_PREREQ_INTTOSTR
   gl_PREREQ_IMAXTOSTR
   gl_PREREQ_OFFTOSTR
   gl_PREREQ_UMAXTOSTR
+  gl_PREREQ_UINTTOSTR
 ])
 
 # Prerequisites of lib/inttostr.h.
 AC_DEFUN([gl_PREREQ_INTTOSTR], [
-  AC_REQUIRE([gl_AC_TYPE_INTMAX_T])
-  AC_REQUIRE([gl_AC_TYPE_UINTMAX_T])
   AC_REQUIRE([AC_TYPE_OFF_T])
   :
 ])
@@ -28,3 +32,6 @@ AC_DEFUN([gl_PREREQ_OFFTOSTR], [:])
 
 # Prerequisites of lib/umaxtostr.c.
 AC_DEFUN([gl_PREREQ_UMAXTOSTR], [:])
+
+# Prerequisites of lib/uinttostr.c.
+AC_DEFUN([gl_PREREQ_UINTTOSTR], [:])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/inttypes_h.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/inttypes_h.m4
