--- dbg/find14/find/gnulib-cvs/gnulib/modules/inttostr	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/inttostr	2020-01-28 23:39:42.000000000 -0600
@@ -7,25 +7,23 @@ lib/inttostr.c
 lib/inttostr.h
 lib/offtostr.c
 lib/umaxtostr.c
-m4/intmax_t.m4
+lib/uinttostr.c
 m4/inttostr.m4
-m4/longlong.m4
-m4/uintmax_t.m4
 
 Depends-on:
+intprops
+stdint
 
 configure.ac:
 gl_INTTOSTR
 
 Makefile.am:
-EXTRA_DIST += inttostr.c
-lib_SOURCES += imaxtostr.c inttostr.h offtostr.c umaxtostr.c
 
 Include:
 "inttostr.h"
 
 License:
-GPL
+LGPL
 
 Maintainer:
 Paul Eggert
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: inttypes
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: inttypes-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: i-ring
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isapipe
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isnanl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isnanl-nolibm
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isnanl-nolibm-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isnanl-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isnan-nolibm
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: isnan-nolibm-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/javacomp dbg/find6/find/gnulib-cvs/gnulib/modules/javacomp
