--- dbg/find14/find/gnulib-cvs/gnulib/lib/inttostr.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/inttostr.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
 /* inttostr.c -- convert integers to printable strings
 
-   Copyright (C) 2001 Free Software Foundation, Inc.
+   Copyright (C) 2001, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -14,10 +14,12 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by Paul Eggert */
 
+#include <config.h>
+
 #include "inttostr.h"
 
 /* Convert I to a printable string in BUF, which must be at least
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/inttostr.h dbg/find6/find/gnulib-cvs/gnulib/lib/inttostr.h
