--- dbg/find14/find/gnulib-cvs/gnulib/lib/javaexec.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/javaexec.h	2020-01-28 23:39:42.000000000 -0600
@@ -14,7 +14,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #ifndef _JAVAEXEC_H
 #define _JAVAEXEC_H
Only in dbg/find14/find/gnulib-cvs/gnulib/lib: javaexec.sh.in
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: javaversion.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: javaversion.class
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: javaversion.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: javaversion.java
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/lbrkprop.h dbg/find6/find/gnulib-cvs/gnulib/lib/lbrkprop.h
