--- dbg/find14/find/gnulib-cvs/gnulib/modules/javaexec	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/javaexec	2020-01-28 23:39:42.000000000 -0600
@@ -4,8 +4,6 @@ Execute a Java program.
 Files:
 lib/javaexec.h
 lib/javaexec.c
-lib/javaexec.sh.in
-m4/javaexec.m4
 
 Depends-on:
 stdbool
@@ -13,19 +11,18 @@ execute
 classpath
 xsetenv
 sh-quote
-pathname
+filename
 xalloc
 xallocsa
 error
-gettext
+gettext-h
+javaexec-script
 
 configure.ac:
-gt_JAVAEXEC
 
 Makefile.am:
 DEFS += -DEXEEXT=\"$(EXEEXT)\"
 lib_SOURCES += javaexec.h javaexec.c
-EXTRA_DIST += javaexec.sh.in
 
 Include:
 "javaexec.h"
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: javaexec-script
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: javaversion
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: lchmod
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/lchown dbg/find6/find/gnulib-cvs/gnulib/modules/lchown
