--- dbg/find14/find/gnulib-cvs/gnulib/lib/lbrkprop.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/lbrkprop.h	2020-01-28 23:39:42.000000000 -0600
@@ -14,8 +14,8 @@ MERCHANTABILITY or FITNESS FOR A PARTICU
 GNU General Public License for more details.
 
 You should have received a copy of the GNU General Public License
-along with this program; if not, write to the Free Software
-Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+along with this program; if not, write to the Free Software Foundation,
+Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #define lbrkprop_header_0 16
 #define lbrkprop_header_1 15
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: lchmod.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/lchown.c dbg/find6/find/gnulib-cvs/gnulib/lib/lchown.c
