--- dbg/find14/find/gnulib-cvs/gnulib/lib/lchown.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/lchown.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,7 @@
 /* Provide a stub lchown function for systems that lack it.
-   Copyright (C) 1998, 1999, 2002, 2004 Free Software Foundation, Inc.
+
+   Copyright (C) 1998, 1999, 2002, 2004, 2006 Free Software
+   Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,22 +15,15 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* written by Jim Meyering */
 
 #include <config.h>
 
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <errno.h>
-
 #include "lchown.h"
-#include "stat-macros.h"
 
-/* Declare chown to avoid a warning.  Don't include unistd.h,
-   because it may have a conflicting prototype for lchown.  */
-int chown ();
+#include <sys/stat.h>
 
 /* Work just like chown, except when FILE is a symbolic link.
    In that case, set errno to EOPNOTSUPP and return -1.
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/lchown.h dbg/find6/find/gnulib-cvs/gnulib/lib/lchown.h
