--- dbg/find14/find/gnulib-cvs/gnulib/modules/lchown	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/lchown	2020-01-28 23:39:42.000000000 -0600
@@ -4,17 +4,16 @@ lchown() function: change ownership of a
 Files:
 lib/lchown.h
 lib/lchown.c
-lib/stat-macros.h
 m4/lchown.m4
 
 Depends-on:
 chown
+sys_stat
 
 configure.ac:
 gl_FUNC_LCHOWN
 
 Makefile.am:
-lib_SOURCES += stat-macros.h
 
 Include:
 "lchown.h"
@@ -24,4 +23,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: ldd
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: ldexpl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: ldexpl-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: lgpl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: lib-ignore
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/linebreak dbg/find6/find/gnulib-cvs/gnulib/modules/linebreak
