--- dbg/find14/find/gnulib-cvs/gnulib/m4/link-follow.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/link-follow.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,8 +1,8 @@
-#serial 5
-dnl Run a program to determine whether whether link(2) follows symlinks.
+#serial 9
+dnl Run a program to determine whether link(2) follows symlinks.
 dnl Set LINK_FOLLOWS_SYMLINKS accordingly.
 
-# Copyright (C) 1999, 2000, 2001, 2004 Free Software Foundation, Inc.
+# Copyright (C) 1999, 2000, 2001, 2004, 2005, 2006 Free Software Foundation, Inc.
 # This file is free software; the Free Software Foundation
 # gives unlimited permission to copy and/or distribute it,
 # with or without modifications, as long as this notice is preserved.
@@ -13,20 +13,14 @@ AC_DEFUN([gl_AC_FUNC_LINK_FOLLOWS_SYMLIN
     [whether link(2) dereferences a symlink specified with a trailing slash],
 		 jm_ac_cv_func_link_follows_symlink,
   [
-    dnl poor-man's AC_REQUIRE: FIXME: repair this once autoconf-3 provides
-    dnl the appropriate framework.
-    test -z "$ac_cv_header_unistd_h" \
-      && AC_CHECK_HEADERS(unistd.h)
-
     # Create a regular file.
     echo > conftest.file
     AC_TRY_RUN(
       [
 #       include <sys/types.h>
 #       include <sys/stat.h>
-#       ifdef HAVE_UNISTD_H
-#        include <unistd.h>
-#       endif
+#       include <unistd.h>
+#       include <stdlib.h>
 
 #       define SAME_INODE(Stat_buf_1, Stat_buf_2) \
 	  ((Stat_buf_1).st_ino == (Stat_buf_2).st_ino \
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/localcharset.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/localcharset.m4
