--- dbg/find14/find/gnulib-cvs/gnulib/m4/localcharset.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/localcharset.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-# localcharset.m4 serial 2
-dnl Copyright (C) 2002 Free Software Foundation, Inc.
+# localcharset.m4 serial 5
+dnl Copyright (C) 2002, 2004, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -7,9 +7,8 @@ dnl with or without modifications, as lo
 AC_DEFUN([gl_LOCALCHARSET],
 [
   dnl Prerequisites of lib/localcharset.c.
-  AC_CHECK_HEADERS_ONCE(stddef.h stdlib.h string.h)
   AC_REQUIRE([AM_LANGINFO_CODESET])
-  AC_CHECK_FUNCS_ONCE(setlocale)
+  AC_CHECK_DECLS_ONCE(getc_unlocked)
 
   dnl Prerequisites of the lib/Makefile.am snippet.
   AC_REQUIRE([AC_CANONICAL_HOST])
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: locale-fr.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: locale-tr.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: locale-zh.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: lock.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/longdouble.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/longdouble.m4
