--- dbg/find14/find/locate/locatedb.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/locate/locatedb.h	2020-01-28 23:39:42.000000000 -0600
@@ -13,7 +13,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software
-   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
    USA.
 */
 
@@ -32,6 +32,19 @@
    This value (which is -128) indicates that the difference is
    too large to fit into one byte, and a two-byte integer follows.  */
 #define	LOCATEDB_ESCAPE 0x80
+#define LOCATEDB_ONEBYTE_MAX (127)
+#define LOCATEDB_ONEBYTE_MIN (-127)
+
+
+/* If it is ever possible to try to encode LOCATEDB_MAGIC as a
+ * single-byte offset, then an unfortunate length of common prefix
+ * will produce a spurious escape character, desynchronising the data
+ * stream.  We use a compile-time check in the preprocessor to prevent
+ * this.
+ */
+#if LOCATEDB_ESCAPE <= LOCATEDB_ONEBYTE_MAX
+#error "You have a bad combination of LOCATEDB_ESCAPE and LOCATEDB_ONEBYTE_MAX, see above"
+#endif
 
 /* These are used for old, bigram-encoded databases:  */
 
@@ -49,4 +62,7 @@
 #  endif
 # endif
 
+
+#define SLOCATE_DB_MAGIC_LEN 2
+
 #endif /* !_LOCATEDB_H */
Binary files dbg/find14/find/locate/locate.o and dbg/find6/find/locate/locate.o differ
diff -up -r dbg/find14/find/locate/Makefile dbg/find6/find/locate/Makefile
