--- dbg/find14/find/gnulib-cvs/gnulib/modules/lstat	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/lstat	2020-01-28 23:39:42.000000000 -0600
@@ -1,13 +1,13 @@
 Description:
-stat() function: return information about a file or symbolic link.
+lstat() function: return information about a file or symbolic link.
 
 Files:
 lib/lstat.c
+lib/lstat.h
 m4/lstat.m4
 
 Depends-on:
-stat
-xalloc
+sys_stat
 
 configure.ac:
 gl_FUNC_LSTAT
@@ -15,11 +15,10 @@ gl_FUNC_LSTAT
 Makefile.am:
 
 Include:
-<sys/stat.h>
+"lstat.h"
 
 License:
-GPL
+LGPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: maintainer-makefile
Only in dbg/find14/find/gnulib-cvs/gnulib/modules: makepath
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/malloc dbg/find6/find/gnulib-cvs/gnulib/modules/malloc
