--- dbg/find14/find/gnulib-cvs/gnulib/modules/mathl	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/mathl	2020-01-28 23:39:42.000000000 -0600
@@ -2,7 +2,6 @@ Description:
 C99 functions for transcendental functions with long double arguments.
 
 Files:
-lib/mathl.h
 lib/acosl.c
 lib/asinl.c
 lib/atanl.c
@@ -10,8 +9,6 @@ lib/ceill.c
 lib/cosl.c
 lib/expl.c
 lib/floorl.c
-lib/frexpl.c
-lib/ldexpl.c
 lib/logl.c
 lib/sincosl.c
 lib/sinl.c
@@ -22,15 +19,19 @@ lib/trigl.h
 m4/mathl.m4
 
 Depends-on:
+math
+frexpl
+isnanl
+ldexpl
 
 configure.ac:
 gl_FUNC_LONG_DOUBLE_MATH
 
 Makefile.am:
-noinst_HEADERS += mathl.h trigl.h trigl.c sincosl.c
+noinst_HEADERS += trigl.h trigl.c sincosl.c
 
 Include:
-"mathl.h"
+<math.h>
 
 License:
 GPL
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: math-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbchar
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbfile
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbiter
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbscasecmp
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbscasecmp-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbscasestr
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbscasestr-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbschr
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbschr-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbscspn
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbscspn-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbslen
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsncasecmp
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsncasecmp-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbspbrk
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbspbrk-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbspcasecmp
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbspcasecmp-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsrchr
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsrchr-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbssep
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsspn
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsspn-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsstr
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbsstr-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbstok_r
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/mbswidth dbg/find6/find/gnulib-cvs/gnulib/modules/mbswidth
