--- dbg/find14/find/gnulib-cvs/gnulib/m4/mathl.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/mathl.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-# mathl.m4 serial 2
-dnl Copyright (c) 2003 Free Software Foundation, Inc.
+# mathl.m4 serial 4
+dnl Copyright (c) 2003, 2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -7,7 +7,7 @@ dnl with or without modifications, as lo
 AC_DEFUN([gl_FUNC_LONG_DOUBLE_MATH], [
 
 AC_CHECK_LIB(m, atan)
-AC_REPLACE_FUNCS(floorl ceill sqrtl frexpl ldexpl asinl acosl atanl \
+AC_REPLACE_FUNCS(floorl ceill sqrtl asinl acosl atanl \
         logl expl tanl sinl cosl)
 
 ])
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: mbchar.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: mbfile.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: mbiter.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/mbrtowc.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/mbrtowc.m4
