--- dbg/find14/find/gnulib-cvs/gnulib/modules/mbswidth	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/mbswidth	2020-01-28 23:39:42.000000000 -0600
@@ -9,6 +9,9 @@ m4/mbrtowc.m4
 m4/mbswidth.m4
 
 Depends-on:
+wchar
+wctype
+wcwidth
 
 configure.ac:
 gl_MBSWIDTH
@@ -24,4 +27,3 @@ GPL
 
 Maintainer:
 Bruno Haible
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mbuiter
Only in dbg/find14/find/gnulib-cvs/gnulib/modules: md5
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/memcasecmp dbg/find6/find/gnulib-cvs/gnulib/modules/memcasecmp
