--- dbg/find14/find/gnulib-cvs/gnulib/lib/mbswidth.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/mbswidth.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* Determine the number of screen columns needed for a string.
-   Copyright (C) 2000-2004 Free Software Foundation, Inc.
+   Copyright (C) 2000-2004, 2007 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,7 +13,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #include <stddef.h>
 
@@ -30,7 +30,7 @@
 #endif
 
 
-#ifdef	__cplusplus
+#ifdef __cplusplus
 extern "C" {
 #endif
 
@@ -56,6 +56,6 @@ extern int mbswidth (const char *string,
 extern int mbsnwidth (const char *buf, size_t nbytes, int flags);
 
 
-#ifdef	__cplusplus
+#ifdef __cplusplus
 }
 #endif
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: mbuiter.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: md2.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: md2.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: md4.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: md4.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/md5.c dbg/find6/find/gnulib-cvs/gnulib/lib/md5.c
