--- demodir/yesterday/mcts.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/mcts.py	2019-11-29 23:50:49.000000000 -0600
@@ -206,9 +164,11 @@ class MCTS:
         # let's release the lock when logits are ready, but probability might not be computed
         node.lock.acquire()
         if not node.probability_ready:
+            # COMMENT THIS OUT TO USE OLD FUNCTION
             for edge in node.edges:
                 edge.value=edge.value.item()
                 edge.logit=edge.logit.item()
+            # COMMENT THIS OUT TO USE OLD FUNCTION
 
             # probability
             prob_tensor = self.nn.logits_to_probability(torch.Tensor([edge.logit for edge in node.edges]))
