--- demodir/yesterday/mcts.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/mcts.py	2019-11-29 23:50:49.000000000 -0600
@@ -154,15 +112,15 @@ class MCTS:
         else:
             maxscore = - float("inf")
             maxedx = None
-            for edx, score in enumerate(visits):
+            for edx, score in enumerate(scores):
                 if score > maxscore:
                     maxedx = edx
                     maxscore = score
-            pi = [0] * len(visits)
+            pi = [0] * len(scores)
             pi[maxedx] = 1
             max_action = root.edges[maxedx]
             self.permaTree.move_root(max_action.to_node)
-        if self.permaTree.last_capture == self.peace:
+        if self.permaTree.last_capture == 40:
             print("Terminated due to peaceful activity")
             return 2
 
