--- demodir/yesterday/mcts.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/mcts.py	2019-11-29 23:50:49.000000000 -0600
@@ -25,9 +25,8 @@ class MCTS:
         # changes to False after the first 30 moves
         self.temperature = True
         self.temperature_change_at = 30
-        self.puct = 20
+        self.puct = 10000
         self.max_game_length = max_game_length
-        self.peace = peace
         self.time_steps = []
         self.is_cuda = is_cuda
         self.simulations_per_play = simulations_per_play
