--- demodir/yesterday/mcts.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/mcts.py	2019-11-29 23:50:49.000000000 -0600
@@ -139,13 +97,13 @@ class MCTS:
         root = self.permaTree.root
         if len(root.edges) == 0:
             return 1
-        visits = []
+        scores = []
         for level_one_edge in root.edges:
             vc = level_one_edge.visit_count
-            visits.append(vc)
+            scores.append(vc)
         if self.temperature:
-            sum_visits = sum(visits)
-            pi = [visit / sum_visits for visit in visits]
+            sum_scores = sum(scores)
+            pi = [score / sum_scores for score in scores]
             # samples instead
 
             sampled_action = random.choices(range(len(root.edges)), weights=pi, k=1)
