--- demodir/yesterday/mcts.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/mcts.py	2019-11-29 23:50:49.000000000 -0600
@@ -290,26 +240,20 @@ class MCTS:
         trace back the whole path from given node till root node while updating edges on the path
 
         :param leaf_node:
-        :param v: the leaf's perspective value, accessed from leaf_node.from_edge
         :return:
         """
         # parent of root node is null
         current_node = leaf_node
-        leaf_first_player=leaf_node.is_first_player()
         while current_node.parent is not None:
             edge = current_node.from_edge
             edge.visit_count = edge.visit_count + 1
-            if edge.to_node.is_first_player()==leaf_first_player:
-                edge.total_action_value = edge.total_action_value - v
-            else:
-                edge.total_action_value = edge.total_action_value + v
+            edge.total_action_value = edge.total_action_value + v
             edge.mean_action_value = edge.total_action_value / edge.visit_count
             current_node = edge.from_node
 
     def print_root(self):
         self.permaTree.root.checker_state.print_board()
 
-
 # interfaces with the Zero
 class TimeStep:
     def __init__(self, checker_state, children_states, mcts_pi):
diff -up -r demodir/yesterday/neuralnetwork.py demodir/today/neuralnetwork.py
