--- dbg/find14/find/gnulib-cvs/gnulib/m4/memcoll.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/memcoll.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,12 +1,13 @@
-# memcoll.m4 serial 4
-dnl Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+# memcoll.m4 serial 7
+dnl Copyright (C) 2002, 2003, 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_MEMCOLL],
 [
+  AC_LIBOBJ([memcoll])
+
   dnl Prerequisites of lib/memcoll.c.
-  AC_REQUIRE([AC_FUNC_MEMCMP])
   AC_FUNC_STRCOLL
 ])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/memmem.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/memmem.m4
