--- dbg/find14/find/gnulib-cvs/gnulib/m4/memcasecmp.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/memcasecmp.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,12 +1,10 @@
-# memcasecmp.m4 serial 2
-dnl Copyright (C) 2002 Free Software Foundation, Inc.
+#serial 5
+dnl Copyright (C) 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_MEMCASECMP],
 [
-  dnl Prerequisites of lib/memcasecmp.c.
-  AC_REQUIRE([AC_HEADER_STDC])
-  AC_CHECK_FUNCS_ONCE(isascii)
+  AC_LIBOBJ([memcasecmp])
 ])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/memcmp.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/memcmp.m4
