--- dbg/find14/find/gnulib-cvs/gnulib/lib/memcpy.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/memcpy.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,4 +1,4 @@
-/* Copyright (C) 1995, 1997, 2000, 2003 Free Software Foundation, Inc.
+/* Copyright (C) 1995, 1997, 2000, 2003, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -12,13 +12,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by Jim Meyering <meyering@na-net.ornl.gov>.  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include <stddef.h>
 
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/memmem.c dbg/find6/find/gnulib-cvs/gnulib/lib/memmem.c
