--- dbg/find14/find/gnulib/lib/mempcpy.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/mempcpy.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* Copy memory area and return pointer after last written byte.
-   Copyright (C) 2003 Free Software Foundation, Inc.
+   Copyright (C) 2003, 2007 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,11 +13,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-/* Specification.  */
-#include "mempcpy.h"
+#include <config.h>
 
+/* Specification.  */
 #include <string.h>
 
 /* Copy N bytes of SRC to DEST, return pointer to bytes after the
Only in dbg/find14/find/gnulib/lib: mempcpy.h
Only in dbg/find6/find/gnulib/lib: memrchr.c
diff -up -r dbg/find14/find/gnulib/lib/memset.c dbg/find6/find/gnulib/lib/memset.c
