--- dbg/find14/find/gnulib-cvs/gnulib/m4/memrchr.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/memrchr.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,16 +1,20 @@
-# memrchr.m4 serial 4
-dnl Copyright (C) 2002, 2003, 2005 Free Software Foundation, Inc.
+# memrchr.m4 serial 7
+dnl Copyright (C) 2002, 2003, 2005, 2006, 2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_FUNC_MEMRCHR],
 [
-  AC_LIBSOURCES([memrchr.c, memrchr.h])
-
   dnl Persuade glibc <string.h> to declare memrchr().
   AC_REQUIRE([AC_GNU_SOURCE])
 
+  AC_REQUIRE([gl_HEADER_STRING_H_DEFAULTS])
+  AC_CHECK_DECLS_ONCE([memrchr])
+  if test $ac_cv_have_decl_memrchr = no; then
+    HAVE_DECL_MEMRCHR=0
+  fi
+
   AC_REPLACE_FUNCS(memrchr)
   if test $ac_cv_func_memrchr = no; then
     gl_PREREQ_MEMRCHR
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: memxor.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: minmax.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: mkancesdirs.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: mkdir-p.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/mkdir-slash.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/mkdir-slash.m4
