--- dbg/find14/find/gnulib-cvs/gnulib/lib/minmax.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/minmax.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* MIN, MAX macros.
-   Copyright (C) 1995, 1998, 2001, 2003 Free Software Foundation, Inc.
+   Copyright (C) 1995, 1998, 2001, 2003, 2005 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,7 +13,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 #ifndef _MINMAX_H
 #define _MINMAX_H
@@ -24,8 +24,15 @@
    #include this file as the last one among the #include list.  */
 
 /* Before we define the following symbols we get the <limits.h> file
-   since otherwise we get redefinitions on some systems.  */
-#include <limits.h>
+   since otherwise we get redefinitions on some systems if <limits.h> is
+   included after this file.  Likewise for <sys/param.h>.
+   If more than one of these system headers define MIN and MAX, pick just
+   one of the headers (because the definitions most likely are the same).  */
+#if HAVE_MINMAX_IN_LIMITS_H
+# include <limits.h>
+#elif HAVE_MINMAX_IN_SYS_PARAM_H
+# include <sys/param.h>
+#endif
 
 /* Note: MIN and MAX should be used with two arguments of the
    same type.  They might not return the minimum and maximum of their two
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: mkancesdirs.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: mkancesdirs.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: mkdirat.c
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/mkdir.c dbg/find6/find/gnulib-cvs/gnulib/lib/mkdir.c
