--- dbg/find14/find/gnulib-cvs/gnulib/modules/modechange	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/modechange	2020-01-28 23:39:42.000000000 -0600
@@ -8,13 +8,15 @@ lib/modechange.c
 m4/modechange.m4
 
 Depends-on:
-xstrtol
+stat-macros
+stdbool
+sys_stat
+xalloc
 
 configure.ac:
 gl_MODECHANGE
 
 Makefile.am:
-lib_SOURCES += modechange.h modechange.c
 
 Include:
 "modechange.h"
@@ -24,4 +26,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/mountlist dbg/find6/find/gnulib-cvs/gnulib/modules/mountlist
