--- dbg/find14/find/gnulib-cvs/gnulib/modules/mountlist	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/mountlist	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 Description:
-Return list of mounted filesystems.
+Return list of mounted file systems.
 
 Files:
 lib/mountlist.h
@@ -9,13 +9,13 @@ m4/fstypename.m4
 m4/mountlist.m4
 
 Depends-on:
+stdbool
 xalloc
 
 configure.ac:
 gl_MOUNTLIST
 
 Makefile.am:
-lib_SOURCES += mountlist.h
 
 Include:
 "mountlist.h"
@@ -25,4 +25,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: mpsort
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/nanosleep dbg/find6/find/gnulib-cvs/gnulib/modules/nanosleep
