--- dbg/find14/find/gnulib-cvs/gnulib/modules/nanosleep	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/nanosleep	2020-01-28 23:39:42.000000000 -0600
@@ -6,7 +6,12 @@ lib/nanosleep.c
 m4/nanosleep.m4
 
 Depends-on:
-timespec
+clock-time
+extensions
+gettime
+stdbool
+sys_time
+time
 
 configure.ac:
 gl_FUNC_NANOSLEEP
@@ -16,9 +21,11 @@ Makefile.am:
 Include:
 <time.h>
 
+Link:
+$(LIB_NANOSLEEP)
+
 License:
 GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: netinet_in
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: netinet_in-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: no-c++
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/obstack dbg/find6/find/gnulib-cvs/gnulib/modules/obstack
