--- dbg/find14/find/gnulib-cvs/gnulib/modules/openat	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/openat	2020-01-28 23:39:42.000000000 -0600
@@ -2,15 +2,28 @@ Description:
 Open a file at a directory.
 
 Files:
+lib/at-func.c
+lib/fchmodat.c
+lib/fchownat.c
+lib/fstatat.c
+lib/mkdirat.c
 lib/openat.c
 lib/openat.h
+lib/openat-die.c
+lib/openat-priv.h
+lib/openat-proc.c
 m4/openat.m4
 
 Depends-on:
+dirname
+extensions
+gettext-h
+intprops
+lchown
+lstat
+same-inode
 save-cwd
-gettext
-error
-exitfail
+stdbool
 
 configure.ac:
 gl_FUNC_OPENAT
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: oset
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: pagealign_alloc
Only in dbg/find14/find/gnulib-cvs/gnulib/modules: path-concat
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/pathmax dbg/find6/find/gnulib-cvs/gnulib/modules/pathmax
