--- dbg/find14/find/gnulib-cvs/gnulib/m4/perl.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/perl.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,11 +1,9 @@
-#serial 7
+#serial 8
 
 dnl From Jim Meyering.
 dnl Find a new-enough version of Perl.
-dnl
 
-# Copyright (C) 1998, 1999, 2000, 2001, 2003, 2004 Free Software
-# Foundation, Inc.
+# Copyright (C) 1998-2001, 2003, 2004, 2007 Free Software Foundation, Inc.
 #
 # This file is free software; the Free Software Foundation
 # gives unlimited permission to copy and/or distribute it,
@@ -13,9 +11,8 @@ dnl
 
 AC_DEFUN([gl_PERL],
 [
-  dnl FIXME: don't hard-code 5.003
-  dnl FIXME: should we cache the result?
-  AC_MSG_CHECKING([for perl5.003 or newer])
+  dnl FIXME: don't hard-code 5.005
+  AC_MSG_CHECKING([for perl5.005 or newer])
   if test "${PERL+set}" = set; then
     # `PERL' is set in the user's environment.
     candidate_perl_names="$PERL"
@@ -31,7 +28,7 @@ AC_DEFUN([gl_PERL],
   for perl in $candidate_perl_names; do
     # Run test in a subshell; some versions of sh will print an error if
     # an executable is not found, even if stderr is redirected.
-    if ( $perl -e 'require 5.003; use File::Compare' ) > /dev/null 2>&1; then
+    if ( $perl -e 'require 5.005; use File::Compare' ) > /dev/null 2>&1; then
       PERL=$perl
       found=yes
       break
@@ -40,7 +37,7 @@ AC_DEFUN([gl_PERL],
 
   AC_MSG_RESULT($found)
   test $found = no && AC_MSG_WARN([
-WARNING: You don't seem to have perl5.003 or newer installed, or you lack
+WARNING: You don't seem to have perl5.005 or newer installed, or you lack
          a usable version of the Perl File::Compare module.  As a result,
          you may be unable to run a few tests or to regenerate certain
          files if you modify the sources from which they are derived.
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/physmem.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/physmem.m4
