--- demodir/yesterday/permatree.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/permatree.py	2019-11-29 23:50:49.000000000 -0600
@@ -143,5 +139,3 @@ class PermaNode:
         #     edge.prior_probability = npp[edx]
 
 
-    def is_first_player(self):
-        return not self.checker_state.flipped
\ No newline at end of file
Only in demodir/yesterday: plots
Binary files demodir/yesterday/__pycache__/checker.cpython-36.pyc and demodir/today/__pycache__/checker.cpython-36.pyc differ
diff -up -r demodir/yesterday/README.md demodir/today/README.md
