--- dbg/find14/find/gnulib-cvs/gnulib/modules/posixver	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/posixver	2020-01-28 23:39:42.000000000 -0600
@@ -13,7 +13,6 @@ configure.ac:
 gl_POSIXVER
 
 Makefile.am:
-lib_SOURCES += posixver.h posixver.c
 
 Include:
 "posixver.h"
@@ -23,4 +22,3 @@ GPL
 
 Maintainer:
 Paul Eggert, Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: printf-frexp
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: printf-frexpl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: printf-frexpl-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: printf-frexp-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: printf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: printf-posix-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/quote dbg/find6/find/gnulib-cvs/gnulib/modules/quote
