--- dbg/find14/find/find/testsuite/find.gnu/printf-symlink.exp	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/find/testsuite/find.gnu/printf-symlink.exp	2020-01-28 23:39:42.000000000 -0600
@@ -2,5 +2,5 @@ exec rm -rf tmp
 exec mkdir tmp
 exec touch tmp/file
 exec ln -s file tmp/LINK
-find_start p {tmp/LINK  -printf "RESULT: %y %Y %p\n" }
+find_start p {tmp/LINK  -printf "RESULT: %y %Y %p\n" -printf "RESULT2: %Y %y %p\n" }
 exec rm -rf tmp
diff -up -r dbg/find14/find/find/testsuite/find.gnu/printf-symlink.xo dbg/find6/find/find/testsuite/find.gnu/printf-symlink.xo
