--- dbg/find14/find/gnulib/lib/quote.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/quote.h	2020-01-28 23:39:42.000000000 -0600
@@ -15,7 +15,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 
 char const *quote_n (int n, char const *name);
Binary files dbg/find14/find/gnulib/lib/quote.o and dbg/find6/find/gnulib/lib/quote.o differ
diff -up -r dbg/find14/find/gnulib/lib/readlink.c dbg/find6/find/gnulib/lib/readlink.c
