--- dbg/find14/find/gnulib-cvs/gnulib/lib/raise.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/raise.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,6 @@
 /* Provide a non-threads replacement for the POSIX raise function.
-   Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+
+   Copyright (C) 2002, 2003, 2005, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -13,7 +14,7 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* written by Jim Meyering */
 
@@ -21,9 +22,7 @@
 
 #include <sys/types.h>
 #include <signal.h>
-#if HAVE_UNISTD_H
-# include <unistd.h>
-#endif
+#include <unistd.h>
 
 int
 raise (int sig)
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: read-file.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: read-file.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: readline.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: readline.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/readlink.c dbg/find6/find/gnulib-cvs/gnulib/lib/readlink.c
