--- dbg/find14/find/gnulib-cvs/gnulib/modules/readutmp	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/readutmp	2020-01-28 23:39:42.000000000 -0600
@@ -8,12 +8,13 @@ m4/readutmp.m4
 
 Depends-on:
 xalloc
+free
+stdbool
 
 configure.ac:
 gl_READUTMP
 
 Makefile.am:
-lib_SOURCES += readutmp.h
 
 Include:
 "readutmp.h"
@@ -22,5 +23,4 @@ License:
 GPL
 
 Maintainer:
-Jim Meyering
-
+Jim Meyering & Paul Eggert
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/realloc dbg/find6/find/gnulib-cvs/gnulib/modules/realloc
