--- dbg/find14/find/gnulib-cvs/gnulib/doc/README	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/doc/README	2020-01-28 23:39:42.000000000 -0600
@@ -1,16 +1 @@
-Sketchy start at information on using files in gnulib.
-
-for mktemp, need these files (including adding to EXTRA_DIST):
-$GNULIBSRC/m4/check-decl.m4	m4
-$GNULIBSRC/m4/mkstemp.m4	m4
-$GNULIBSRC/m4/prereq.m4		m4
-$GNULIBSRC/lib/tempname.c	lib
-$GNULIBSRC/lib/mkstemp.c	lib
-
-then in configure.ac:
-AC_DEFUN([texinfo_MACROS],
-[
-  AC_REQUIRE([gl_PREREQ_TEMPNAME])dnl for mkstemp
-  AC_REQUIRE([UTILS_FUNC_MKSTEMP])
-])dnl
-texinfo_MACROS
+regexprops-generic.texi is generated via a utility in findutils.
Only in dbg/find6/find/gnulib-cvs/gnulib/doc: regexprops-generic.texi
Only in dbg/find6/find/gnulib-cvs/gnulib/doc: relocatable-maint.texi
Only in dbg/find6/find/gnulib-cvs/gnulib/doc: relocatable.texi
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/doc/solaris-versions dbg/find6/find/gnulib-cvs/gnulib/doc/solaris-versions
