--- dbg/find14/find/gnulib-cvs/gnulib/modules/regex	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/regex	2020-01-28 23:39:42.000000000 -0600
@@ -4,18 +4,28 @@ Regular expression matching.
 Files:
 lib/regex.h
 lib/regex.c
+lib/regex_internal.c
+lib/regex_internal.h
+lib/regexec.c
+lib/regcomp.c
 m4/regex.m4
 
 Depends-on:
 alloca
-gettext
-restrict
+extensions
+gettext-h
+localcharset
+malloc
+stdbool
+stdint
+ssize_t
+wchar
+wctype
 
 configure.ac:
 gl_REGEX
 
 Makefile.am:
-lib_SOURCES += regex.h
 
 Include:
 "regex.h"
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: regexprops-generic
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: relocatable-lib
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: relocatable-lib-lgpl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: relocatable-prog
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: relocatable-prog-wrapper
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: relocatable-script
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/rename dbg/find6/find/gnulib-cvs/gnulib/modules/rename
