--- dbg/find14/find/gnulib-cvs/gnulib/modules/rename	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/rename	2020-01-28 23:39:42.000000000 -0600
@@ -10,7 +10,7 @@ xalloc
 dirname
 
 configure.ac:
-vb_FUNC_RENAME
+gl_FUNC_RENAME
 
 Makefile.am:
 
@@ -22,4 +22,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: rename-dest-slash
Only in dbg/find14/find/gnulib-cvs/gnulib/modules: restrict
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/rmdir dbg/find6/find/gnulib-cvs/gnulib/modules/rmdir
