--- dbg/find14/find/gnulib-cvs/gnulib/lib/rename.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/rename.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,8 @@
 /* Work around the bug in some systems whereby rename fails when the source
-   path has a trailing slash.  The rename functions of SunOS 4.1.1_U1 and
+   file has a trailing slash.  The rename functions of SunOS 4.1.1_U1 and
    mips-dec-ultrix4.4 have this bug.
-   Copyright (C) 2001-2003 Free Software Foundation, Inc.
+
+   Copyright (C) 2001, 2002, 2003, 2005, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -15,13 +16,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* written by Volker Borchert */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 #undef rename
 
 #include <stdio.h>
@@ -31,27 +30,27 @@
 #include "dirname.h"
 #include "xalloc.h"
 
-/* Rename the file SRC_PATH to DST_PATH, removing any trailing
-   slashes from SRC_PATH.  Needed for SunOS 4.1.1_U1.  */
+/* Rename the file SRC to DST, removing any trailing
+   slashes from SRC.  Needed for SunOS 4.1.1_U1.  */
 
 int
-rpl_rename (const char *src_path, const char *dst_path)
+rpl_rename (char const *src, char const *dst)
 {
   char *src_temp;
   int ret_val;
-  size_t s_len = strlen (src_path);
+  size_t s_len = strlen (src);
 
-  if (s_len && src_path[s_len - 1] == '/')
+  if (s_len && src[s_len - 1] == '/')
     {
-      src_temp = xstrdup (src_path);
+      src_temp = xstrdup (src);
       strip_trailing_slashes (src_temp);
     }
   else
-    src_temp = (char *) src_path;
+    src_temp = (char *) src;
 
-  ret_val = rename (src_temp, dst_path);
+  ret_val = rename (src_temp, dst);
 
-  if (src_temp != src_path)
+  if (src_temp != src)
     free (src_temp);
 
   return ret_val;
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: rename-dest-slash.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: rijndael-alg-fst.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: rijndael-alg-fst.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: rijndael-api-fst.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: rijndael-api-fst.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/rmdir.c dbg/find6/find/gnulib-cvs/gnulib/lib/rmdir.c
