--- dbg/find14/find/gnulib-cvs/gnulib/modules/rmdir	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/rmdir	2020-01-28 23:39:42.000000000 -0600
@@ -3,16 +3,15 @@ rmdir() function: delete a directory.
 
 Files:
 lib/rmdir.c
-lib/stat-macros.h
 m4/rmdir.m4
 
 Depends-on:
+sys_stat
 
 configure.ac:
 gl_FUNC_RMDIR
 
 Makefile.am:
-lib_SOURCES += stat-macros.h
 
 Include:
 <unistd.h>
@@ -22,4 +21,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: rmdir-errno
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/rpmatch dbg/find6/find/gnulib-cvs/gnulib/modules/rpmatch
