--- dbg/find14/find/xargs/testsuite/xargs.posix/s6.exp	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/xargs/testsuite/xargs.posix/s6.exp	2020-01-28 23:39:42.000000000 -0600
@@ -1 +1 @@
-xargs_start f {-s6} files.xi
+xargs_start f {-s6} files.xi skip
Only in dbg/find6/find/xargs/testsuite/xargs.posix: space.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: space-I.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: space.xo
Only in dbg/find6/find/xargs/testsuite/xargs.posix: sv-bug-18714b.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: sv-bug-18714b.xo
Only in dbg/find6/find/xargs/testsuite/xargs.posix: sv-bug-18714.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: sv-bug-18714.xo
Only in dbg/find6/find/xargs/testsuite/xargs.posix: uc_L2.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: uc_L2.xo
Only in dbg/find6/find/xargs/testsuite/xargs.posix: unmatched2.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: unmatched2.xo
Only in dbg/find6/find/xargs/testsuite/xargs.posix: unmatched.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: unmatched-n2-x.exp
Only in dbg/find6/find/xargs/testsuite/xargs.posix: unmatched.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: e.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: empty_def-t.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: empty_def-t.xe
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: empty_def-t.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: empty-t.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: empty-t.xe
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: empty-t.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: eof1.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: eof1.xo
Only in dbg/find14/find/xargs/testsuite/xargs.sysv: eof.exp
Only in dbg/find14/find/xargs/testsuite/xargs.sysv: eof.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: e.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: idef.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: idef-s26.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: idef-s26.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: idef.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: l1_2.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: l1_2.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: l1.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: l1.xo
Only in dbg/find14/find/xargs/testsuite/xargs.sysv: l2.exp
Only in dbg/find14/find/xargs/testsuite/xargs.sysv: l2.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: lc_l2.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: lc_l2.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: s30-t.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: s30-t.xe
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: s30-t.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: space-t.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: space-t.xe
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: space-t.xo
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: sv-bug-18713.exp
Only in dbg/find6/find/xargs/testsuite/xargs.sysv: sv-bug-18713.xo
Binary files dbg/find14/find/xargs/xargs and dbg/find6/find/xargs/xargs differ
diff -up -r dbg/find14/find/xargs/xargs.1 dbg/find6/find/xargs/xargs.1
