--- dbg/find14/find/gnulib-cvs/gnulib/modules/safe-read	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/safe-read	2020-01-28 23:39:42.000000000 -0600
@@ -5,15 +5,14 @@ Files:
 lib/safe-read.h
 lib/safe-read.c
 m4/safe-read.m4
-m4/ssize_t.m4
 
 Depends-on:
+ssize_t
 
 configure.ac:
 gl_SAFE_READ
 
 Makefile.am:
-lib_SOURCES += safe-read.h safe-read.c
 
 Include:
 "safe-read.h"
@@ -23,4 +22,3 @@ GPL
 
 Maintainer:
 Jim Meyering, Bruno Haible
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/safe-write dbg/find6/find/gnulib-cvs/gnulib/modules/safe-write
