--- dbg/find14/find/gnulib-cvs/gnulib/modules/safe-write	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/safe-write	2020-01-28 23:39:42.000000000 -0600
@@ -13,7 +13,6 @@ configure.ac:
 gl_SAFE_WRITE
 
 Makefile.am:
-lib_SOURCES += safe-write.h safe-write.c
 
 Include:
 "safe-write.h"
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/same dbg/find6/find/gnulib-cvs/gnulib/modules/same
