--- dbg/find14/find/gnulib-cvs/gnulib/modules/same	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/same	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 Description:
-Determine whether two pathnames refer to the same directory entry of the same
+Determine whether two file names refer to the same directory entry of the same
 directory.
 
 Files:
@@ -11,13 +11,13 @@ Depends-on:
 xalloc
 error
 dirname
+same-inode
 stdbool
 
 configure.ac:
 gl_SAME
 
 Makefile.am:
-lib_SOURCES += same.h same.c
 
 Include:
 "same.h"
@@ -27,4 +27,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: same-inode
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/save-cwd dbg/find6/find/gnulib-cvs/gnulib/modules/save-cwd
