--- dbg/find14/find/gnulib-cvs/gnulib/modules/savedir	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/savedir	2020-01-28 23:39:42.000000000 -0600
@@ -7,13 +7,13 @@ lib/savedir.c
 m4/savedir.m4
 
 Depends-on:
+openat
 xalloc
 
 configure.ac:
 gl_SAVEDIR
 
 Makefile.am:
-lib_SOURCES += savedir.h savedir.c
 
 Include:
 "savedir.h"
@@ -23,4 +23,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: savewd
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: search
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: search-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/setenv dbg/find6/find/gnulib-cvs/gnulib/modules/setenv
