--- dbg/find14/find/gnulib-cvs/gnulib/lib/setenv.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/setenv.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,4 +1,4 @@
-/* Copyright (C) 1992,1995-1999,2000-2004 Free Software Foundation, Inc.
+/* Copyright (C) 1992,1995-1999,2000-2003,2005-2007 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    This program is free software; you can redistribute it and/or modify
@@ -13,9 +13,9 @@
 
    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA. */
 
-#if HAVE_CONFIG_H
+#if !_LIBC
 # include <config.h>
 #endif
 #include <alloca.h>
@@ -31,6 +31,8 @@
 # include <unistd.h>
 #endif
 
+#if _LIBC || !HAVE_SETENV
+
 #if !_LIBC
 # include "allocsa.h"
 #endif
@@ -326,3 +328,5 @@ text_set_element (__libc_subfreeres, fre
 weak_alias (__setenv, setenv)
 weak_alias (__clearenv, clearenv)
 #endif
+
+#endif /* _LIBC || !HAVE_SETENV */
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/setenv.h dbg/find6/find/gnulib-cvs/gnulib/lib/setenv.h
