--- dbg/find14/find/gnulib-cvs/gnulib/modules/setenv	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/setenv	2020-01-28 23:39:42.000000000 -0600
@@ -11,12 +11,13 @@ m4/setenv.m4
 Depends-on:
 allocsa
 alloca-opt
+unistd
 
 configure.ac:
-gt_FUNC_SETENV
+gl_FUNC_SETENV
+gl_FUNC_UNSETENV
 
 Makefile.am:
-lib_SOURCES += setenv.h
 
 Include:
 "setenv.h"
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/settime dbg/find6/find/gnulib-cvs/gnulib/modules/settime
