--- dbg/find14/find/gnulib-cvs/gnulib/m4/setenv.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/setenv.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,16 +1,30 @@
-# setenv.m4 serial 5
-dnl Copyright (C) 2001-2004 Free Software Foundation, Inc.
+# setenv.m4 serial 8
+dnl Copyright (C) 2001-2004, 2006-2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
-AC_DEFUN([gt_FUNC_SETENV],
+AC_DEFUN([gl_FUNC_SETENV],
 [
-  AC_REPLACE_FUNCS(setenv unsetenv)
+  AC_CHECK_FUNCS_ONCE([setenv])
   if test $ac_cv_func_setenv = no; then
+    AC_LIBOBJ([setenv])
     gl_PREREQ_SETENV
   fi
+])
+
+# Like gl_FUNC_SETENV, except prepare for separate compilation (no AC_LIBOBJ).
+AC_DEFUN([gl_FUNC_SETENV_SEPARATE],
+[
+  AC_CHECK_FUNCS_ONCE([setenv])
+  gl_PREREQ_SETENV
+])
+
+AC_DEFUN([gl_FUNC_UNSETENV],
+[
+  AC_CHECK_FUNCS([unsetenv])
   if test $ac_cv_func_unsetenv = no; then
+    AC_LIBOBJ([unsetenv])
     gl_PREREQ_UNSETENV
   else
     AC_CACHE_CHECK([for unsetenv() return type], gt_cv_func_unsetenv_ret,
@@ -57,7 +71,6 @@ AC_DEFUN([gl_PREREQ_SETENV],
   AC_CHECK_HEADERS_ONCE(unistd.h)
   AC_CHECK_HEADERS(search.h)
   AC_CHECK_FUNCS(tsearch)
-  gt_CHECK_VAR_DECL([#include <errno.h>], errno)
   gt_CHECK_VAR_DECL([#include <unistd.h>], environ)
 ])
 
@@ -65,6 +78,5 @@ AC_DEFUN([gl_PREREQ_SETENV],
 AC_DEFUN([gl_PREREQ_UNSETENV],
 [
   AC_CHECK_HEADERS_ONCE(unistd.h)
-  gt_CHECK_VAR_DECL([#include <errno.h>], errno)
   gt_CHECK_VAR_DECL([#include <unistd.h>], environ)
 ])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/settime.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/settime.m4
