--- dbg/find14/find/gnulib-cvs/gnulib/lib/sh-quote.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/sh-quote.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* Shell quoting.
-   Copyright (C) 2001-2004 Free Software Foundation, Inc.
+   Copyright (C) 2001-2004, 2006 Free Software Foundation, Inc.
    Written by Bruno Haible <haible@clisp.cons.org>, 2001.
 
    This program is free software; you can redistribute it and/or modify
@@ -14,11 +14,9 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 /* Specification.  */
 #include "sh-quote.h"
@@ -90,7 +88,7 @@ shell_quote_argv (char **argv)
 	    break;
 	}
 
-      command = (char *) xmalloc (length);
+      command = XNMALLOC (length, char);
 
       p = command;
       for (argp = argv; ; )
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/sh-quote.h dbg/find6/find/gnulib-cvs/gnulib/lib/sh-quote.h
