--- dbg/find14/find/gnulib-cvs/gnulib/m4/sha1.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/sha1.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,11 +1,13 @@
-# sha1.m4 serial 5
-dnl Copyright (C) 2002, 2003, 2004 Free Software Foundation, Inc.
+# sha1.m4 serial 7
+dnl Copyright (C) 2002, 2003, 2004, 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_SHA1],
 [
+  AC_LIBOBJ([sha1])
+
   dnl Prerequisites of lib/sha1.c.
   AC_REQUIRE([AC_C_BIGENDIAN])
   :
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/sig2str.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/sig2str.m4
