--- dbg/find14/find/gnulib-cvs/gnulib/lib/sig2str.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/sig2str.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
 /* sig2str.h -- convert between signal names and numbers
 
-   Copyright (C) 2002 Free Software Foundation, Inc.
+   Copyright (C) 2002, 2005 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -14,19 +14,16 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by Paul Eggert.  */
 
-/* Include <signal.h> before including this file.  */
+#include <signal.h>
 
 /* Don't override system declarations of SIG2STR_MAX, sig2str, str2sig.  */
 #ifndef SIG2STR_MAX
 
-/* Upper bound on the string length of an integer converted to string.
-   302 / 1000 is ceil (log10 (2.0)).  Subtract 1 for the sign bit;
-   add 1 for integer division truncation; add 1 more for a minus sign.  */
-# define INT_STRLEN_BOUND(t) ((sizeof (t) * CHAR_BIT - 1) * 302 / 1000 + 2)
+# include "intprops.h"
 
 /* Size of a buffer needed to hold a signal name like "HUP".  */
 # define SIG2STR_MAX (sizeof "SIGRTMAX" + INT_STRLEN_BOUND (int) - 1)
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: sigprocmask.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: sigprocmask.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/sincosl.c dbg/find6/find/gnulib-cvs/gnulib/lib/sincosl.c
