--- dbg/find14/find/gnulib-cvs/gnulib/m4/sig2str.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/sig2str.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,11 +1,12 @@
-# sig2str.m4 serial 2
-dnl Copyright (C) 2002 Free Software Foundation, Inc.
+#serial 5
+dnl Copyright (C) 2002, 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_FUNC_SIG2STR],
 [
+
   AC_REPLACE_FUNCS(sig2str)
   if test $ac_cv_func_sig2str = no; then
     gl_PREREQ_SIG2STR
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/signalblocking.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/signalblocking.m4
