--- dbg/find14/find/gnulib-cvs/gnulib/lib/sinl.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/sinl.c	2020-01-28 23:39:42.000000000 -0600
@@ -13,6 +13,11 @@
  * ====================================================
  */
 
+#include <config.h>
+
+/* Specification.  */
+#include <math.h>
+
 /* sinl(x)
  * Return sine function of x.
  *
@@ -44,13 +49,10 @@
  *	TRIG(x) returns trig(x) nearly rounded
  */
 
-#include <math.h>
-
-#include "mathl.h"
-
 #include "trigl.h"
 #include "trigl.c"
 #include "sincosl.c"
+#include "isnanl.h"
 
 long double
 sinl (long double x)
@@ -58,13 +60,17 @@ sinl (long double x)
   long double y[2], z = 0.0L;
   int n;
 
+  /* sinl(NaN) is NaN */
+  if (isnanl (x))
+    return x;
+
   /* |x| ~< pi/4 */
   if (x >= -0.7853981633974483096156608458198757210492 &&
       x <= 0.7853981633974483096156608458198757210492)
     return kernel_sinl (x, z, 0);
 
-    /* sinl(Inf or NaN) is NaN, sinl(0) is 0 */
-  else if (x + x == x || x != x)
+    /* sinl(Inf) is NaN, sinl(0) is 0 */
+  else if (x + x == x)
     return x - x;		/* NaN */
 
   /* argument reduction needed */
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: size_max.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/snprintf.c dbg/find6/find/gnulib-cvs/gnulib/lib/snprintf.c
