--- dbg/find14/find/gnulib-cvs/gnulib/modules/snprintf	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/snprintf	2020-01-28 23:39:42.000000000 -0600
@@ -2,25 +2,24 @@ Description:
 snprintf() function: print formatted output to a fixed length string
 
 Files:
-lib/snprintf.h
 lib/snprintf.c
 m4/snprintf.m4
 
 Depends-on:
+stdio
 vasnprintf
-minmax
 
 configure.ac:
 gl_FUNC_SNPRINTF
+gl_STDIO_MODULE_INDICATOR([snprintf])
 
 Makefile.am:
-lib_SOURCES += snprintf.h
 
 Include:
-"snprintf.h"
+<stdio.h>
 
 License:
 LGPL
 
 Maintainer:
-Simon Josefsson
+Simon Josefsson, Paul Eggert, Bruno Haible
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: snprintf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: snprintf-posix-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: snprintf-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: socklen
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: sprintf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: sprintf-posix-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: ssize_t
Only in dbg/find14/find/gnulib-cvs/gnulib/modules: stat
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: stat-macros
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: stat-time
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: stat-time-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: stdarg
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/stdbool dbg/find6/find/gnulib-cvs/gnulib/modules/stdbool
