--- dbg/find14/find/gnulib-cvs/gnulib/config/srclistvars.sh	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/config/srclistvars.sh	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,25 @@
-# $Id: srclistvars.sh,v 1.22 2004/12/12 14:39:22 karl Exp $
+# $Id: srclistvars.sh,v 1.28 2006/08/22 19:38:57 eggert Exp $
 # Variables for srclist-update and srclist.txt.
 # Will change for each user.
 
+# Copyright (C) 2002, 2003, 2004 2005, 2006 Free Software Foundation,
+# Inc.
+
+# This file is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street - Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
 case $LOGNAME in
 jas | eggert)
   : ${AUTOCONF=../autoconf}
@@ -18,20 +36,20 @@ jas | eggert)
 
 karl)
   : ${ACLOCAL=/usr/local/gnu/share/aclocal}
-  : ${AUTOCONF=$HOME/gnu/src/autoconf}
-  : ${AUTOMAKE=$HOME/gnu/src/automake}
-  : ${EMACSSRC=$HOME/gnu/src/emacs}
-  : ${GETTEXT=$HOME/gnu/src/gettext}
   : ${GNUBIN=/usr/local/gnu/bin}
-  : ${GNUCONFIG=$HOME/gnu/src/config}
-  : ${GNULIBSRC=$HOME/gnu/src/gnulib}
-  : ${GNUORG=$HOME/gnu/gnuorg}
-  : ${GNUSTANDARDS=$HOME/gnu/src/gnustandards}
   : ${GNUWWWLICENSES=$HOME/gnu/www/www/licenses}
-  : ${LIBCSRC=$HOME/gnu/src/libc}
-  : ${LIBTOOL=$HOME/gnu/src/libtool}
-  : ${TEXINFOSRC=/u/texinfo/src}
   : ${TEXMFROOT=/usr/local/texmf/texmf-dist}
+  : ${AUTOCONF=../autoconf}
+  : ${AUTOMAKE=../automake}
+  : ${EMACSSRC=../emacs}
+  : ${GETTEXT=../gettext}
+  : ${GNUCONFIG=../config}
+  : ${GNUORG=$HOME/gnu/gnuorg}
+  : ${GNULIBSRC=../gnulib}
+  : ${GNUSTANDARDS=../gnustandards}
+  : ${LIBCSRC=../libc}
+  : ${LIBTOOL=../libtool}
+  : ${TEXINFOSRC=../texinfo}
   ;;
 
 esac
Only in dbg/find14/find/gnulib-cvs/gnulib/config: texinfo.tex
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/COPYING dbg/find6/find/gnulib-cvs/gnulib/COPYING
