--- dbg/find14/find/gnulib-cvs/gnulib/config/srclist-update	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/config/srclist-update	2020-01-28 23:39:42.000000000 -0600
@@ -1,12 +1,10 @@
 #!/bin/sh
-# $Id: srclist-update,v 1.15 2004/06/22 18:28:25 eggert Exp $
+# $Id: srclist-update,v 1.19 2007/03/30 23:44:27 karl Exp $
 #
 # Check for files in directory $1 being up to date, according to the
 # list on stdin.  Don't actually make any changes, just show the diffs.
 #
-# Source `dirname $0`/srclistvars.sh first, if it exists.
-
-# Copyright (C) 2002, 2003 Free Software Foundation, Inc.
+# Copyright (C) 2002, 2003, 2005, 2007 Free Software Foundation, Inc.
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -19,11 +17,10 @@
 # GNU General Public License for more details.
 #
 # You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
-# 02111-1307, USA.
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
-# Written by Karl Berry.
+# Originally written by Karl Berry.
 
 
 if test -n "$1"; then
@@ -36,6 +33,7 @@ verbose=false
 : ${TMPDIR=/tmp}
 dsttmp=$TMPDIR/srclist.dst
 
+# Source `dirname $0`/srclistvars.sh first, if it exists.
 mydir=`dirname $0`
 test -r $mydir/srclistvars.sh && . $mydir/srclistvars.sh
 
@@ -56,7 +54,7 @@ fixlicense='
 #\
 #   You should have received a copy of the GNU General Public License along\
 #   with this program; if not, write to the Free Software Foundation,\
-#   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
   /Th[ei][ s].* is free software/,/\*\//c\
    This program is free software; you can redistribute it and/or modify\
@@ -71,11 +69,11 @@ fixlicense='
 \
    You should have received a copy of the GNU General Public License along\
    with this program; if not, write to the Free Software Foundation,\
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 '
 
 # sed command to remove lines containing $Id lines.
-# Quote the $Id so that CVS does not expand it in this script.
+# Quote the $ so that CVS does not expand it in this script.
 remove_id_lines='/[$]Id:.*[$]/d'
 
 # $1 is input file, $2 is output.
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/config/srclistvars.sh dbg/find6/find/gnulib-cvs/gnulib/config/srclistvars.sh
