--- dbg/find14/find/gnulib-cvs/gnulib/modules/stdbool	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/stdbool	2020-01-28 23:39:42.000000000 -0600
@@ -13,13 +13,14 @@ AM_STDBOOL_H
 
 Makefile.am:
 BUILT_SOURCES += $(STDBOOL_H)
-EXTRA_DIST += stdbool_.h
 
-# We need the following in order to create an <stdbool.h> when the system
+# We need the following in order to create <stdbool.h> when the system
 # doesn't have one that works.
-all-local $(lib_OBJECTS): $(STDBOOL_H)
 stdbool.h: stdbool_.h
-	sed -e 's/@''HAVE__BOOL''@/$(HAVE__BOOL)/g' < $(srcdir)/stdbool_.h > $@-t
+	rm -f $@-t $@
+	{ echo '/* DO NOT EDIT! GENERATED AUTOMATICALLY! */'; \
+	  sed -e 's/@''HAVE__BOOL''@/$(HAVE__BOOL)/g' < $(srcdir)/stdbool_.h; \
+	} > $@-t
 	mv $@-t $@
 MOSTLYCLEANFILES += stdbool.h stdbool.h-t
 
@@ -31,4 +32,3 @@ LGPL
 
 Maintainer:
 Bruno Haible
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: stdbool-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/stdint dbg/find6/find/gnulib-cvs/gnulib/modules/stdint
