--- dbg/find14/find/gnulib-cvs/gnulib/m4/stdio-safer.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/stdio-safer.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,11 +1,15 @@
-# stdio-safer.m4 serial 2
-dnl Copyright (C) 2002 Free Software Foundation, Inc.
+#serial 10
+dnl Copyright (C) 2002, 2005-2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
-AC_DEFUN([gl_STDIO_SAFER],
+AC_DEFUN([gl_FOPEN_SAFER],
 [
-  dnl Prerequisites of lib/fopen-safer.c.
-  AC_CHECK_HEADERS_ONCE(unistd.h)
+  AC_LIBOBJ([fopen-safer])
+])
+
+AC_DEFUN([gl_TMPFILE_SAFER],
+[
+  AC_LIBOBJ([tmpfile-safer])
 ])
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: stdlib_h.m4
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: stdlib-safer.m4
Only in dbg/find14/find/gnulib-cvs/gnulib/m4: st_mtim.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/stpcpy.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/stpcpy.m4
