--- dbg/find14/find/gnulib-cvs/gnulib/lib/stdio-safer.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/stdio-safer.h	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,6 @@
 /* Invoke stdio functions, but avoid some glitches.
 
-   Copyright (C) 2001, 2003 Free Software Foundation, Inc.
+   Copyright (C) 2001, 2003, 2006 Free Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -14,10 +14,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* Written by Paul Eggert.  */
 
 #include <stdio.h>
 
 FILE *fopen_safer (char const *, char const *);
+FILE *tmpfile_safer (void);
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: stdlib_.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: stdlib--.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: stdlib-safer.h
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/stpcpy.c dbg/find6/find/gnulib-cvs/gnulib/lib/stpcpy.c
