--- dbg/find14/find/gnulib/lib/stpcpy.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/lib/stpcpy.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* stpcpy.c -- copy a string and return pointer to end of new string
-   Copyright (C) 1992, 1995, 1997, 1998 Free Software Foundation, Inc.
+   Copyright (C) 1992, 1995, 1997-1998, 2006 Free Software Foundation, Inc.
 
    NOTE: The canonical source of this file is maintained with the GNU C Library.
    Bugs can be reported to bug-glibc@prep.ai.mit.edu.
@@ -15,18 +15,17 @@
    GNU General Public License for more details.
 
    You should have received a copy of the GNU General Public License
-   along with this program; if not, write to the Free Software
-   Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
-   USA.  */
+   along with this program; if not, write to the Free Software Foundation,
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #include <string.h>
 
 #undef __stpcpy
-#undef stpcpy
+#ifdef _LIBC
+# undef stpcpy
+#endif
 
 #ifndef weak_alias
 # define __stpcpy stpcpy
Only in dbg/find14/find/gnulib/lib: stpcpy.h
diff -up -r dbg/find14/find/gnulib/lib/strdup.c dbg/find6/find/gnulib/lib/strdup.c
