--- dbg/find14/find/gnulib-cvs/gnulib/modules/stpcpy	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/stpcpy	2020-01-28 23:39:42.000000000 -0600
@@ -2,24 +2,23 @@ Description:
 stpcpy() function: copy a string, returning a pointer to its end.
 
 Files:
-lib/stpcpy.h
 lib/stpcpy.c
 m4/stpcpy.m4
 
 Depends-on:
+string
 
 configure.ac:
 gl_FUNC_STPCPY
+gl_STRING_MODULE_INDICATOR([stpcpy])
 
 Makefile.am:
-lib_SOURCES += stpcpy.h
 
 Include:
-"stpcpy.h"
+<string.h>
 
 License:
 LGPL
 
 Maintainer:
 Bruno Haible, glibc
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/stpncpy dbg/find6/find/gnulib-cvs/gnulib/modules/stpncpy
