--- dbg/find14/find/gnulib-cvs/gnulib/m4/stpncpy.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/stpncpy.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-# stpncpy.m4 serial 2
-dnl Copyright (C) 2002-2003, 2005 Free Software Foundation, Inc.
+# stpncpy.m4 serial 6
+dnl Copyright (C) 2002-2003, 2005-2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -9,6 +9,8 @@ AC_DEFUN([gl_FUNC_STPNCPY],
   dnl Persuade glibc <string.h> to declare stpncpy().
   AC_REQUIRE([AC_GNU_SOURCE])
 
+  AC_REQUIRE([gl_HEADER_STRING_H_DEFAULTS])
+
   dnl Both glibc and AIX (4.3.3, 5.1) have an stpncpy() function
   dnl declared in <string.h>. Its side effects are the same as those
   dnl of strncpy():
@@ -22,6 +24,8 @@ AC_DEFUN([gl_FUNC_STPNCPY],
   AC_CACHE_CHECK([for working stpncpy], gl_cv_func_stpncpy, [
     AC_TRY_RUN([
 #include <stdlib.h>
+#include <string.h> /* for strcpy */
+/* The stpncpy prototype is missing in <string.h> on AIX 4.  */
 extern char *stpncpy (char *dest, const char *src, size_t n);
 int main () {
   const char *src = "Hello";
@@ -49,8 +53,7 @@ int main () {
     AC_DEFINE(HAVE_STPNCPY, 1,
       [Define if you have the stpncpy() function and it works.])
   else
-    AC_DEFINE([stpncpy], [gnu_stpncpy],
-      [Define to a replacement function name for stpncpy().])
+    HAVE_STPNCPY=0
     AC_LIBOBJ([stpncpy])
     gl_PREREQ_STPNCPY
   fi
@@ -60,4 +63,3 @@ int main () {
 AC_DEFUN([gl_PREREQ_STPNCPY], [
   :
 ])
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/strcase.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/strcase.m4
