--- dbg/find14/find/gnulib-cvs/gnulib/modules/stpncpy	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/stpncpy	2020-01-28 23:39:42.000000000 -0600
@@ -2,24 +2,23 @@ Description:
 stpncpy() function: copy a size-bounded string, returning a pointer to its end.
 
 Files:
-lib/stpncpy.h
 lib/stpncpy.c
 m4/stpncpy.m4
 
 Depends-on:
+string
 
 configure.ac:
 gl_FUNC_STPNCPY
+gl_STRING_MODULE_INDICATOR([stpncpy])
 
 Makefile.am:
-lib_SOURCES += stpncpy.h
 
 Include:
-"stpncpy.h"
+<string.h>
 
 License:
 LGPL
 
 Maintainer:
 Bruno Haible, glibc
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/strcase dbg/find6/find/gnulib-cvs/gnulib/modules/strcase
