--- dbg/find14/find/gnulib-cvs/gnulib/m4/strchrnul.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/strchrnul.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-# strchrnul.m4 serial 3
-dnl Copyright (C) 2003 Free Software Foundation, Inc.
+# strchrnul.m4 serial 5
+dnl Copyright (C) 2003, 2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -9,8 +9,10 @@ AC_DEFUN([gl_FUNC_STRCHRNUL],
   dnl Persuade glibc <string.h> to declare strchrnul().
   AC_REQUIRE([AC_GNU_SOURCE])
 
+  AC_REQUIRE([gl_HEADER_STRING_H_DEFAULTS])
   AC_REPLACE_FUNCS(strchrnul)
   if test $ac_cv_func_strchrnul = no; then
+    HAVE_STRCHRNUL=0
     gl_PREREQ_STRCHRNUL
   fi
 ])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/strdup.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/strdup.m4
