--- dbg/find14/find/gnulib-cvs/gnulib/lib/strerror.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/strerror.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,7 +1,7 @@
 /* strerror.c --- ANSI C compatible system error routine
 
-   Copyright (C) 1986, 1988, 1989, 1991, 2002, 2003 Free Software
-   Foundation, Inc.
+   Copyright (C) 1986, 1988, 1989, 1991, 2002, 2003, 2006, 2007 Free
+   Software Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -15,11 +15,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
-#if HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
+
+#if !HAVE_STRERROR
 
 #include <limits.h>
 
@@ -47,3 +47,11 @@ strerror (int n)
   else
     return sys_errlist[n];
 }
+
+#else
+
+/* This declaration is solely to ensure that after preprocessing
+   this file is never empty.  */
+typedef int dummy;
+
+#endif
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/strftime.c dbg/find6/find/gnulib-cvs/gnulib/lib/strftime.c
