--- dbg/find14/find/gnulib-cvs/gnulib/modules/strndup	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/strndup	2020-01-28 23:39:42.000000000 -0600
@@ -2,25 +2,24 @@ Description:
 strndup() function: duplicate a size-bounded string.
 
 Files:
-lib/strndup.h
 lib/strndup.c
 m4/strndup.m4
 
 Depends-on:
+string
 strnlen
 
 configure.ac:
 gl_FUNC_STRNDUP
+gl_STRING_MODULE_INDICATOR([strndup])
 
 Makefile.am:
-lib_SOURCES += strndup.h
 
 Include:
-"strndup.h"
+<string.h>
 
 License:
 LGPL
 
 Maintainer:
 glibc
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/strnlen dbg/find6/find/gnulib-cvs/gnulib/modules/strnlen
