--- dbg/find14/find/gnulib-cvs/gnulib/modules/strpbrk	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/strpbrk	2020-01-28 23:39:42.000000000 -0600
@@ -2,24 +2,23 @@ Description:
 strpbrk() function: search a string for any of a set of characters.
 
 Files:
-lib/strpbrk.h
 lib/strpbrk.c
 m4/strpbrk.m4
 
 Depends-on:
+string
 
 configure.ac:
 gl_FUNC_STRPBRK
+gl_STRING_MODULE_INDICATOR([strpbrk])
 
 Makefile.am:
-lib_SOURCES += strpbrk.h
 
 Include:
-"strpbrk.h"
+<string.h>
 
 License:
 LGPL
 
 Maintainer:
 Bruno Haible, glibc
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: strptime
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/strsep dbg/find6/find/gnulib-cvs/gnulib/modules/strsep
