--- dbg/find14/find/gnulib-cvs/gnulib/modules/strtoimax	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/strtoimax	2020-01-28 23:39:42.000000000 -0600
@@ -4,16 +4,17 @@ strtoimax() function: convert string to
 Files:
 lib/strtoimax.c
 m4/longlong.m4
-m4/stdint_h.m4
-m4/inttypes_h.m4
-m4/intmax_t.m4
 m4/strtoimax.m4
 
 Depends-on:
 strtoll
+verify
+inttypes
+stdint
 
 configure.ac:
 gl_FUNC_STRTOIMAX
+gl_INTTYPES_MODULE_INDICATOR([strtoimax])
 
 Makefile.am:
 
@@ -24,4 +25,3 @@ GPL
 
 Maintainer:
 Paul Eggert
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/strtok_r dbg/find6/find/gnulib-cvs/gnulib/modules/strtok_r
