--- dbg/find14/find/gnulib/m4/strtoull.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/m4/strtoull.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
-# strtoull.m4 serial 3
-dnl Copyright (C) 2002, 2004 Free Software Foundation, Inc.
+# strtoull.m4 serial 4
+dnl Copyright (C) 2002, 2004, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -7,9 +7,9 @@ dnl with or without modifications, as lo
 AC_DEFUN([gl_FUNC_STRTOULL],
 [
   dnl We don't need (and can't compile) the replacement strtoull
-  dnl unless the type 'unsigned long long' exists.
-  AC_REQUIRE([gl_AC_TYPE_UNSIGNED_LONG_LONG])
-  if test "$ac_cv_type_unsigned_long_long" = yes; then
+  dnl unless the type 'unsigned long long int' exists.
+  AC_REQUIRE([AC_TYPE_UNSIGNED_LONG_LONG_INT])
+  if test "$ac_cv_type_unsigned_long_long_int" = yes; then
     AC_REPLACE_FUNCS(strtoull)
     if test $ac_cv_func_strtoull = no; then
       gl_PREREQ_STRTOULL
diff -up -r dbg/find14/find/gnulib/m4/strtoul.m4 dbg/find6/find/gnulib/m4/strtoul.m4
