--- dbg/find14/find/gnulib-cvs/gnulib/config/CVS/Tag	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/config/CVS/Tag	2020-01-28 23:39:42.000000000 -0600
@@ -1 +1 @@
-D2005.02.12.00.00.00
+D2007.04.03.00.00.00
Only in dbg/find14/find/gnulib-cvs/gnulib/config: depcomp
Only in dbg/find14/find/gnulib-cvs/gnulib/config: install-sh
Only in dbg/find14/find/gnulib-cvs/gnulib/config: mdate-sh
Only in dbg/find14/find/gnulib-cvs/gnulib/config: missing
Only in dbg/find14/find/gnulib-cvs/gnulib/config: mkinstalldirs
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/config/srclist.txt dbg/find6/find/gnulib-cvs/gnulib/config/srclist.txt
