--- demodir/yesterday/test_checker.py	2020-03-28 20:51:37.174154259 -0500
+++ demodir/today/test_checker.py	2020-03-28 20:51:19.569984012 -0500
@@ -17,8 +17,8 @@ class TestChecker(TestCase):
         print(checker.state.get_legal_actions())
         # player_play_verbose(checker.state)
         original_state=checker.state
-        checker.flip_board()
-        checker.flip_board()
+        checker.flip_board().flip_board()
+
         assert checker.state==original_state
 
 
Only in demodir/yesterday: training_timesteps
Only in demodir/yesterday: validation_timesteps
diff -up -r demodir/yesterday/zero.py demodir/today/zero.py
