--- dbg/find14/test/test.sh	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/test/test.sh	2020-01-28 23:39:42.000000000 -0600
@@ -123,12 +123,15 @@ die() {
 }
 
 TDIR=`mktemp -d`
-ln -s . $TDIR/symlink
+cd $TDIR
+ln -s a b
+ln -s b a
 
 fail=0
-printf "$TDIR\n$TDIR/symlink\n" > exp
-$versiondir/find/find $TDIR -follow | head -c 200 > out || exit 255
-compare exp out || fail=1
+$versiondir/find/find -L -type f
+if test $? -eq 134; then
+  fail=1
+fi
 
 #if [ $fail -eq 1 ]; then
 #  fail=0
@@ -136,8 +139,8 @@ compare exp out || fail=1
 #  fail=1
 #fi
 
+cd -
 rm -rf $TDIR
-
 exit $fail
 
 
