--- demodir/yesterday/test_alphaBetaNode.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/test_alphaBetaNode.py	2019-11-29 23:50:49.000000000 -0600
@@ -18,7 +18,7 @@ class TestAlphaBetaNode(TestCase):
 
         alpha.make_children()
         for child in alpha.children:
-            self.assertTrue(child.first_player_evaluate() == val)
+            self.assertTrue(child.evaluate()== val)
 
     def test_game(self):
         ab = AlphaBeta(4)
diff -up -r demodir/yesterday/test_checker.py demodir/today/test_checker.py
