--- dbg/find14/find/gnulib-cvs/gnulib/tests/test-memmem.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/tests/test-memmem.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2004 Free Software Foundation
+ * Copyright (C) 2004, 2007 Free Software Foundation
  * Written by Simon Josefsson
  *
  * This program is free software; you can redistribute it and/or modify
@@ -14,8 +14,8 @@
  *
  * You should have received a copy of the GNU General Public License
  * along with this program; if not, write to the Free Software
- * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
- * 02111-1307, USA.  */
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.  */
 
 #if HAVE_CONFIG_H
 # include <config.h>
@@ -23,7 +23,6 @@
 
 #include <stdio.h>
 #include <string.h>
-#include "memmem.h"
 
 int
 main (int argc, char *argv[])
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-netinet_in.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-printf-frexp.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-printf-frexpl.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-printf-posix.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-printf-posix.h
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-printf-posix.sh
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-rbtreehash_list.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-rbtree_list.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-rbtree_oset.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-read-file.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-rijndael.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-search.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-snprintf.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-snprintf-posix.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-snprintf-posix.h
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-sprintf-posix.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-sprintf-posix.h
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-stat-time.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-stdbool.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-stdint.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-stdio.c
Only in dbg/find6/find/gnulib-cvs/gnulib/tests: test-stdlib.c
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/tests/test-stpncpy.c dbg/find6/find/gnulib-cvs/gnulib/tests/test-stpncpy.c
