--- dbg/find14/find/THANKS	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/THANKS	2020-01-28 23:39:42.000000000 -0600
@@ -9,7 +9,7 @@ Paul Slootman <paul@murphy.nl>
 Lionel CONS <lionel.cons@cern.ch>
 "Gerrit P. Haase" <gerrit.haase@t-online.de>
 Bruno Haible
-Bob Prolux
+Bob Proulx
 Martin Buchholz
 Steve Revilak 
 Solar Designer <solar@openwall.com>
@@ -19,7 +19,11 @@ Andreas Metzler
 Bas van Gompel 
 Jim Meyering
 Dmitry V. Levin
+Aaron S. Hawley
 
 Suggestion for locate's ignore-case code:
 Matt Mueller <donut@azstarnet.com>
 "Stephane Barizien" <sba@ocegr.fr>
+
+Suggestion for the -sparse predicate:
+dean gaudet <dean@arctic.org>
diff -up -r dbg/find14/find/xargs/.deps/xargs.Po dbg/find6/find/xargs/.deps/xargs.Po
