--- dbg/find14/find/gnulib-cvs/gnulib/m4/timegm.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/timegm.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,11 +1,12 @@
-# timegm.m4 serial 3
-dnl Copyright (C) 2003 Free Software Foundation, Inc.
+# timegm.m4 serial 5
+dnl Copyright (C) 2003, 2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_FUNC_TIMEGM],
 [
+  AC_REQUIRE([gl_HEADER_TIME_H_DEFAULTS])
   AC_REQUIRE([gl_FUNC_MKTIME])
   if test $ac_cv_func_working_mktime = no; then
     # Assume that timegm is buggy if mktime is.
@@ -14,11 +15,13 @@ AC_DEFUN([gl_FUNC_TIMEGM],
   else
     AC_REPLACE_FUNCS(timegm)
   fi
-  if test $ac_cv_func_timegm = no; then
+  REPLACE_TIMEGM=1
+  if test $ac_cv_func_timegm = yes; then
+    AC_CHECK_DECLS([timegm], [REPLACE_TIMEGM=0], [], [#include <time.h>])
+  fi
+  if test $REPLACE_TIMEGM = 1; then
     gl_PREREQ_TIMEGM
   fi
-
-  AC_CHECK_DECLS([timegm], , , [#include <time.h>])
 ])
 
 # Prerequisites of lib/timegm.c.
Only in dbg/find6/find/gnulib-cvs/gnulib/m4: time_h.m4
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/time_r.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/time_r.m4
