--- dbg/find14/find/gnulib-cvs/gnulib/modules/time_r	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/time_r	2020-01-28 23:39:42.000000000 -0600
@@ -3,25 +3,22 @@ Reentrant time functions like localtime_
 
 Files:
 lib/time_r.c
-lib/time_r.h
 m4/time_r.m4
 
 Depends-on:
 extensions
-restrict
+time
 
 configure.ac:
 gl_TIME_R
 
 Makefile.am:
-lib_SOURCES += time_r.h
 
 Include:
-"time_r.h"
+<time.h>
 
 License:
-GPL
+LGPL
 
 Maintainer:
 Paul Eggert
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/timespec dbg/find6/find/gnulib-cvs/gnulib/modules/timespec
