--- dbg/find14/find/gnulib-cvs/gnulib/lib/trigl.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/trigl.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,5 +1,5 @@
 /* Quad-precision floating point argument reduction.
-   Copyright (C) 1999 Free Software Foundation, Inc.
+   Copyright (C) 1999, 2007 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Jakub Jelinek <jj@ultra.linux.cz>
 
@@ -15,12 +15,14 @@
 
    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
+#include <config.h>
+
+/* Specification.  */
 #include <math.h>
-#include <float.h>
 
-#include "mathl.h"
+#include <float.h>
 
 /* Table of constants for 2/pi, 5628 hexadecimal digits of 2/pi */
 static const int two_over_pi[] = {
@@ -231,7 +233,7 @@ ieee754_rem_pio2l (long double x, long d
 	return -1;
       }
 
-  if (x + x == x || x != x)	/* x is +=oo or NaN */
+  if (x + x == x)	/* x is ±oo */
     {
       y[0] = x - x;
       y[1] = y[0];
@@ -421,7 +423,7 @@ static const double PIo2[] = {
 static const double zero = 0.0, one = 1.0, two24 = 1.67772160000000000000e+07,	/* 0x41700000, 0x00000000 */
   twon24 = 5.96046447753906250000e-08;	/* 0x3E700000, 0x00000000 */
 
-int
+static int
 kernel_rem_pio2 (double *x, double *y, int e0, int nx, int prec,
 		 const int *ipio2)
 {
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/trigl.h dbg/find6/find/gnulib-cvs/gnulib/lib/trigl.h
