--- dbg/find14/find/gnulib-cvs/gnulib/lib/trigl.h	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/trigl.h	2020-01-28 23:39:42.000000000 -0600
@@ -17,7 +17,7 @@
 
    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 extern int ieee754_rem_pio2l (long double x, long double *y);
 extern long double kernel_sinl (long double x, long double y, int iy);
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: trim.c
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: trim.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: tsearch.c
Only in dbg/find14/find/gnulib-cvs/gnulib/lib: ucs4-utf16.h
Only in dbg/find14/find/gnulib-cvs/gnulib/lib: ucs4-utf8.h
Only in dbg/find6/find/gnulib-cvs/gnulib/lib: uinttostr.c
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/unicodeio.c dbg/find6/find/gnulib-cvs/gnulib/lib/unicodeio.c
