--- dbg/find14/find/gnulib-cvs/gnulib/modules/unistd-safer	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/unistd-safer	2020-01-28 23:39:42.000000000 -0600
@@ -2,8 +2,11 @@ Description:
 File descriptor functions that avoid clobbering STD{IN,OUT,ERR}_FILENO.
 
 Files:
-lib/unistd-safer.h
+lib/unistd--.h
 lib/dup-safer.c
+lib/fd-safer.c
+lib/pipe-safer.c
+lib/unistd-safer.h
 m4/unistd-safer.m4
 
 Depends-on:
@@ -12,7 +15,6 @@ configure.ac:
 gl_UNISTD_SAFER
 
 Makefile.am:
-lib_SOURCES += unistd-safer.h dup-safer.c
 
 Include:
 "unistd-safer.h"
@@ -22,4 +24,3 @@ GPL
 
 Maintainer:
 Paul Eggert, Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: unistd-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: unistr
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: unitypes
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: unlink-busy
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: unlinkdir
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/unlocked-io dbg/find6/find/gnulib-cvs/gnulib/modules/unlocked-io
