--- dbg/find14/find/locate/updatedb	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/locate/updatedb	2020-01-28 23:39:42.000000000 -0600
@@ -1,6 +1,7 @@
 #! /bin/sh
 # updatedb -- build a locate pathname database
-# Copyright (C) 1994 Free Software Foundation, Inc.
+# Copyright (C) 1994, 1996, 1997, 2000, 2001, 2003, 2004, 2005, 2006
+# Free Software Foundation, Inc.
 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -14,11 +15,14 @@
 
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
 # USA.
 
 # csh original by James Woods; sh conversion by David MacKenzie.
 
+#exec 2> /tmp/updatedb-trace.txt 
+#set -x
+
 usage="\
 Usage: $0 [--findoptions='-option1 -option2...']
        [--localpaths='dir1 dir2...'] [--netpaths='dir1 dir2...']
@@ -48,7 +52,7 @@ do
     --localuser) LOCALUSER="$val" ;;
     --old-format) old=yes ;;
     --changecwd)  changeto="$val" ;;
-    --version) echo "GNU updatedb version 4.2.17"; exit 0 ;;
+    --version) echo "GNU updatedb version 4.3.3-CVS"; exit 0 ;;
     --help) echo "$usage"; exit 0 ;;
     *) echo "updatedb: invalid option $opt
 $usage" >&2
@@ -82,6 +86,24 @@ getuid() {
     id | cut -d'(' -f 1 | cut -d'=' -f2
 }
 
+# figure out if su supports the -s option
+select_shell() {
+    if su "$1" -s $SHELL -c false < /dev/null  ; then
+	# No.
+	echo ""
+    else
+	if su "$1" -s $SHELL -c true < /dev/null  ; then
+	    # Yes.
+	    echo "-s $SHELL"
+        else
+	    # su is unconditionally failing.  We won't be able to 
+	    # figure out what is wrong, so be conservative.
+	    echo ""
+	fi
+    fi
+}
+
+
 # You can set these in the environment, or use command-line options,
 # to override their defaults:
 
@@ -100,6 +122,16 @@ getuid() {
 # Directories to not put in the database, which would otherwise be.
 : ${PRUNEPATHS="/tmp /usr/tmp /var/tmp /afs /amd /sfs"}
 
+# Trailing slashes result in regex items that are never matched, which 
+# is not what the user will expect.   Therefore we now reject such 
+# constructs.
+for p in $PRUNEPATHS; do
+    case "$p" in
+	/*/)   echo "$0: $p: pruned paths should not contain trailing slashes" >&2
+	       exit 1
+    esac
+done
+
 # The same, in the form of a regex that find can use.
 test -z "$PRUNEREGEX" &&
   PRUNEREGEX=`echo $PRUNEPATHS|sed -e 's,^,\\\(^,' -e 's, ,$\\\)\\\|\\\(^,g' -e 's,$,$\\\),'`
@@ -141,6 +173,21 @@ fi
 : ${code:=${LIBEXECDIR}/code}
 
 
+checkbinary () {
+    if test -x "$1" ; then
+	: ok
+    else
+      eval echo "updatedb needs to be able to execute $1, but cannot." >&2
+      exit 1
+    fi
+}
+
+for binary in $find $frcode $bigram $code
+do
+  checkbinary $binary
+done
+
+
 PATH=/bin:/usr/bin:${BINDIR}; export PATH
 
 : ${PRUNEFS="nfs NFS proc afs proc smbfs autofs iso9660 ncpfs coda devpts ftpfs devfs mfs sysfs shfs"}
@@ -166,7 +213,7 @@ cd "$changeto"
 if test -n "$SEARCHPATHS"; then
   if [ "$LOCALUSER" != "" ]; then
     # : A1
-    su $LOCALUSER -s $SHELL -c \
+    su $LOCALUSER `select_shell $LOCALUSER` -c \
     "$find $SEARCHPATHS $FINDOPTIONS \
      \\( $prunefs_exp \
      -type d -regex '$PRUNEREGEX' \\) -prune -o $print_option"
@@ -182,7 +229,7 @@ if test -n "$NETPATHS"; then
 myuid=`getuid` 
 if [ "$myuid" = 0 ]; then
     # : A3
-    su $NETUSER -s $SHELL -c \
+    su $NETUSER `select_shell $NETUSER` -c \
      "$find $NETPATHS $FINDOPTIONS \\( -type d -regex '$PRUNEREGEX' -prune \\) -o $print_option" ||
     exit $?
   else
@@ -193,7 +240,7 @@ if [ "$myuid" = 0 ]; then
 fi
 } | $sort -f | $frcode $frcode_options > $LOCATE_DB.n
 then
-    # OK so far
+    : OK so far
     true
 else
     rv=$?
@@ -236,7 +283,7 @@ cd "$changeto"
 if test -n "$SEARCHPATHS"; then
   if [ "$LOCALUSER" != "" ]; then
     # : A5
-    su $LOCALUSER -s $SHELL -c \
+    su $LOCALUSER `select_shell $LOCALUSER` -c \
     "$find $SEARCHPATHS $FINDOPTIONS \
      \( $prunefs_exp \
      -type d -regex '$PRUNEREGEX' \) -prune -o $print_option" || exit $?
@@ -252,7 +299,7 @@ if test -n "$NETPATHS"; then
   myuid=`getuid`
   if [ "$myuid" = 0 ]; then
     # : A7
-    su $NETUSER -s $SHELL -c \
+    su $NETUSER `select_shell $NETUSER` -c \
      "$find $NETPATHS $FINDOPTIONS \\( -type d -regex '$PRUNEREGEX' -prune \\) -o $print_option" ||
     exit $?
   else
diff -up -r dbg/find14/find/locate/updatedb.1 dbg/find6/find/locate/updatedb.1
