--- dbg/find14/find/gnulib-cvs/gnulib/modules/userspec	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/userspec	2020-01-28 23:39:42.000000000 -0600
@@ -2,6 +2,7 @@ Description:
 Parse a `user:group' specifier (e.g. the first argument of chown utility).
 
 Files:
+lib/inttostr.h
 lib/userspec.c
 lib/userspec.h
 m4/userspec.m4
@@ -11,13 +12,14 @@ posixver
 xalloc
 xstrtol
 strdup
-gettext
+gettext-h
+intprops
+stdbool
 
 configure.ac:
 gl_USERSPEC
 
 Makefile.am:
-lib_SOURCES += userspec.c userspec.h
 
 Include:
 "userspec.h"
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/utf16-ucs4 dbg/find6/find/gnulib-cvs/gnulib/modules/utf16-ucs4
