--- dbg/find14/find/gnulib-cvs/gnulib/lib/utime.c	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/lib/utime.c	2020-01-28 23:39:42.000000000 -0600
@@ -1,4 +1,5 @@
-/* Copyright (C) 1998, 2001, 2002, 2003, 2004 Free Software Foundation, Inc.
+/* Copyright (C) 1998, 2001, 2002, 2003, 2004, 2006 Free Software
+   Foundation, Inc.
 
    This program is free software; you can redistribute it and/or modify it
    under the terms of the GNU General Public License as published by the
@@ -12,13 +13,11 @@
 
    You should have received a copy of the GNU General Public License
    along with this program; if not, write to the Free Software Foundation,
-   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.  */
+   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.  */
 
 /* derived from a function in touch.c */
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 #undef utime
 
 #include <sys/types.h>
@@ -48,6 +47,11 @@ struct utimbuf
 };
 #endif
 
+/* The results of open() in this file are not used with fchdir,
+   therefore save some unnecessary work in fchdir.c.  */
+#undef open
+#undef close
+
 /* Emulate utime (file, NULL) for systems (like 4.3BSD) that do not
    interpret it to set the access and modification times of FILE to
    the current time.  Return 0 if successful, -1 if not. */
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/lib/utimecmp.c dbg/find6/find/gnulib-cvs/gnulib/lib/utimecmp.c
