--- dbg/find14/find/gnulib-cvs/gnulib/m4/utimens.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/m4/utimens.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,14 +1,15 @@
-dnl Copyright (C) 2003, 2004 Free Software Foundation, Inc.
+dnl Copyright (C) 2003, 2004, 2005, 2006, 2007 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_UTIMENS],
 [
+  AC_LIBOBJ([utimens])
+
   dnl Prerequisites of lib/utimens.c.
-  AC_REQUIRE([gl_TIMESPEC])
   AC_REQUIRE([gl_FUNC_UTIMES])
   AC_REQUIRE([gl_CHECK_TYPE_STRUCT_TIMESPEC])
   AC_REQUIRE([gl_CHECK_TYPE_STRUCT_UTIMBUF])
-  AC_CHECK_FUNCS_ONCE(futimes)
+  AC_CHECK_FUNCS_ONCE([futimes futimesat])
 ])
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/m4/utimes.m4 dbg/find6/find/gnulib-cvs/gnulib/m4/utimes.m4
