--- dbg/find14/find/gnulib-cvs/gnulib/modules/utimens	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/utimens	2020-01-28 23:39:42.000000000 -0600
@@ -2,19 +2,20 @@ Description:
 Set file access and modification times.
 
 Files:
-lib/utimens.h
 lib/utimens.c
+lib/utimens.h
+m4/utimbuf.m4
 m4/utimens.m4
 m4/utimes.m4
 
 Depends-on:
-timespec
+sys_time
+time
 
 configure.ac:
 gl_UTIMENS
 
 Makefile.am:
-lib_SOURCES += utimens.h utimens.c
 
 Include:
 "utimens.h"
@@ -24,4 +25,3 @@ GPL
 
 Maintainer:
 Paul Eggert, Jim Meyering
-
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vararrays
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/vasnprintf dbg/find6/find/gnulib-cvs/gnulib/modules/vasnprintf
