--- dbg/find14/find/gnulib-cvs/gnulib/modules/vasprintf	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/vasprintf	2020-01-28 23:39:42.000000000 -0600
@@ -2,22 +2,22 @@ Description:
 vsprintf with automatic memory allocation.
 
 Files:
-lib/vasprintf.h
 lib/vasprintf.c
 lib/asprintf.c
 m4/vasprintf.m4
 
 Depends-on:
 vasnprintf
+stdio
 
 configure.ac:
 gl_FUNC_VASPRINTF
+gl_STDIO_MODULE_INDICATOR([vasprintf])
 
 Makefile.am:
-lib_SOURCES += vasprintf.h
 
 Include:
-"vasprintf.h"
+<stdio.h>
 
 License:
 LGPL
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vasprintf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vasprintf-posix-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vasprintf-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: verify
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: verror
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/version-etc dbg/find6/find/gnulib-cvs/gnulib/modules/version-etc
