--- dbg/find14/find/gnulib-cvs/gnulib/modules/version-etc	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/version-etc	2020-01-28 23:39:42.000000000 -0600
@@ -6,7 +6,8 @@ lib/version-etc.h
 lib/version-etc.c
 
 Depends-on:
-gettext
+gettext-h
+stdarg
 
 configure.ac:
 
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vfprintf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vfprintf-posix-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: visibility
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vprintf-posix
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: vprintf-posix-tests
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/vsnprintf dbg/find6/find/gnulib-cvs/gnulib/modules/vsnprintf
