--- dbg/find14/find/gnulib-cvs/gnulib/modules/wait-process	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/wait-process	2020-01-28 23:39:42.000000000 -0600
@@ -12,8 +12,9 @@ fatal-signal
 error
 exit
 xalloc
-gettext
+gettext-h
 stdbool
+unistd
 
 configure.ac:
 gl_WAIT_PROCESS
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: wchar
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: wchar-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: wctype
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: wctype-tests
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: wcwidth
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: winsz-ioctl
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: winsz-termios
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: write-any-file
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/xalloc dbg/find6/find/gnulib-cvs/gnulib/modules/xalloc
