--- dbg/find14/find/gnulib-cvs/gnulib/modules/xalloc-die	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/xalloc-die	2020-01-28 23:39:42.000000000 -0600
@@ -7,7 +7,7 @@ lib/xalloc-die.c
 Depends-on:
 xalloc
 error
-gettext
+gettext-h
 exitfail
 
 configure.ac:
@@ -23,4 +23,3 @@ GPL
 
 Maintainer:
 all
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/xgetcwd dbg/find6/find/gnulib-cvs/gnulib/modules/xgetcwd
