--- dbg/find14/find/gnulib/m4/xgetcwd.m4	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib/m4/xgetcwd.m4	2020-01-28 23:39:42.000000000 -0600
@@ -1,10 +1,12 @@
-# xgetcwd.m4 serial 4
-dnl Copyright (C) 2002, 2003, 2004 Free Software Foundation, Inc.
+#serial 6
+dnl Copyright (C) 2002, 2003, 2004, 2005, 2006 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
 
 AC_DEFUN([gl_XGETCWD],
 [
+  AC_LIBOBJ([xgetcwd])
+
   AC_REQUIRE([gl_FUNC_GETCWD])
 ])
Only in dbg/find14/find/gnulib/m4: xreadlink.m4
Only in dbg/find6/find/gnulib/m4: xstrndup.m4
Only in dbg/find6/find/gnulib/m4: xstrtod.m4
diff -up -r dbg/find14/find/gnulib/m4/xstrtol.m4 dbg/find6/find/gnulib/m4/xstrtol.m4
