--- dbg/find14/find/gnulib-cvs/gnulib/modules/xreadlink	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/xreadlink	2020-01-28 23:39:42.000000000 -0600
@@ -1,21 +1,20 @@
 Description:
-Return the contents of a symbolic link, without size limitations.
+Reading symbolic links without size limitation.
 
 Files:
 lib/xreadlink.h
 lib/xreadlink.c
-m4/xreadlink.m4
-m4/ssize_t.m4
 
 Depends-on:
-xalloc
 readlink
+ssize_t
+unistd
+xalloc
 
 configure.ac:
-gl_XREADLINK
 
 Makefile.am:
-lib_SOURCES += xreadlink.h xreadlink.c
+lib_SOURCES += xreadlink.c
 
 Include:
 "xreadlink.h"
@@ -24,5 +23,4 @@ License:
 GPL
 
 Maintainer:
-Jim Meyering
-
+Bruno Haible, Jim Meyering
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: xreadlink-with-size
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/xsetenv dbg/find6/find/gnulib-cvs/gnulib/modules/xsetenv
