--- dbg/find14/find/gnulib-cvs/gnulib/modules/xsize	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/xsize	2020-01-28 23:39:42.000000000 -0600
@@ -4,9 +4,9 @@ Checked size_t computations.
 Files:
 lib/xsize.h
 m4/xsize.m4
-m4/size_max.m4
 
 Depends-on:
+size_max
 
 configure.ac:
 gl_XSIZE
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: xstriconv
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/xstrtod dbg/find6/find/gnulib-cvs/gnulib/modules/xstrtod
