--- dbg/find14/find/gnulib-cvs/gnulib/modules/xstrtol	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/xstrtol	2020-01-28 23:39:42.000000000 -0600
@@ -1,27 +1,23 @@
 Description:
-Convert string to 'long` or 'unsigned long', with error checking.
+Convert string to 'long' or 'unsigned long', with error checking.
 
 Files:
 lib/xstrtol.h
 lib/xstrtol.c
 lib/xstrtoul.c
-m4/ulonglong.m4
-m4/longlong.m4
-m4/stdint_h.m4
-m4/inttypes_h.m4
-m4/uintmax_t.m4
-m4/intmax_t.m4
 m4/xstrtol.m4
 
 Depends-on:
 exitfail
 error
+gettext-h
+intprops
+inttypes
 
 configure.ac:
 gl_XSTRTOL
 
 Makefile.am:
-lib_SOURCES += xstrtol.h xstrtol.c xstrtoul.c
 
 Include:
 "xstrtol.h"
Only in dbg/find6/find/gnulib-cvs/gnulib/modules: xstrtold
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/xstrtoumax dbg/find6/find/gnulib-cvs/gnulib/modules/xstrtoumax
