--- dbg/find14/find/gnulib-cvs/gnulib/modules/xstrtod	2020-01-28 23:47:42.000000000 -0600
+++ dbg/find6/find/gnulib-cvs/gnulib/modules/xstrtod	2020-01-28 23:39:42.000000000 -0600
@@ -7,12 +7,12 @@ lib/xstrtod.c
 m4/xstrtod.m4
 
 Depends-on:
+stdbool
 
 configure.ac:
 gl_XSTRTOD
 
 Makefile.am:
-lib_SOURCES += xstrtod.h xstrtod.c
 
 Include:
 "xstrtod.h"
@@ -22,4 +22,3 @@ GPL
 
 Maintainer:
 Jim Meyering
-
diff -up -r dbg/find14/find/gnulib-cvs/gnulib/modules/xstrtoimax dbg/find6/find/gnulib-cvs/gnulib/modules/xstrtoimax
