--- demodir/yesterday/zero.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/zero.py	2019-11-29 23:50:49.000000000 -0600
@@ -533,25 +427,9 @@ def gpu_thread_worker(nn, edge_queue, ev
 
             if last_batch:
                 edge_queue.task_done()
-                print("Queue task done signal sent. Queue will join. Thread may still be running.")
+                # print("Queue task done signal sent. Queue will join. Thread may still be running.")
                 return
 
-
-def mcts_search_worker(nn_thread_edge_queue, nn, is_cuda, max_game_length, peace, simulations_per_play,
-                       debug, epoch, new_time_steps):
-    mcts = MCTS(nn_thread_edge_queue, nn, is_cuda,
-                max_game_length, peace, simulations_per_play,
-                debug)
-    mcts.play_until_terminal()
-    new_time_steps += mcts.time_steps
-
 if __name__ == '__main__':
-    az = AlphaZero("highpuctshortgame28", is_cuda=False)
+    az = AlphaZero("alpha", is_cuda=True)
     az.train()
-
-# fork alternate is longer games, value coef = 1
-# longer game does not seem to work. All the games now look the same? The loss diverges?
-
-# high puct highpuctshortgame28
-# low puct = 1/2
-# alternate =4
