--- demodir/yesterday/zero.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/zero.py	2019-11-29 23:50:49.000000000 -0600
@@ -512,18 +394,30 @@ def gpu_thread_worker(nn, edge_queue, ev
                 states = [edge.to_node.checker_state for edge in edges]
                 input_tensor = states_to_batch_tensor(states, is_cuda)
                 # this line is the bottleneck
-                if isinstance(nn, YesPolicy) or isinstance(nn, SharedPolicy):
+                if isinstance(nn, YesPolicy):
                     value_tensor, logits_tensor = nn(input_tensor)
 
                 else:
                     value_tensor = nn(input_tensor)
 
-                if isinstance(nn, YesPolicy) or isinstance(nn, SharedPolicy):
-                    logits_tensor = value_tensor
+                if isinstance(nn, YesPolicy):
+                    logits_tensor=value_tensor
+
+                # value
+                # value_array = value_tensor.cpu().numpy()
+                # value_array = np.squeeze(value_array, axis=1)
+                # value_list = value_array.tolist()
+                # if isinstance(nn, YesPolicy):
+                #     # logits
+                #     logit_array = logits_tensor.cpu().numpy()
+                #     logit_array = np.squeeze(logit_array, axis=1)
+                #     logit_list = logit_array.tolist()
+                # else:
+                #     logit_list = value_list
 
                 for edx, edge in enumerate(edges):
-                    edge.value = value_tensor[edx, 0]
-                    edge.logit = logits_tensor[edx, 0]
+                    edge.value = value_tensor[edx,0]
+                    edge.logit = logits_tensor[edx,0]
                     edge_queue.task_done()
                     edge.from_node.unassigned -= 1
                     if edge.from_node.unassigned == 0:
