--- demodir/yesterday/zero.py	2019-12-13 16:22:23.000000000 -0600
+++ demodir/today/zero.py	2019-11-29 23:50:49.000000000 -0600
@@ -499,7 +381,7 @@ def gpu_thread_worker(nn, edge_queue, ev
                     edge = edge_queue.get_nowait()
                     if edge is None:
                         last_batch = True
-                        print("Sentinel received. GPU will process this batch and terminate afterwards")
+                        # print("Sentinel received. GPU will process this batch and terminate afterwards")
                     else:
                         edges.append(edge)
                 except queue.Empty:
