--- yesterday/tmp/combined	2020-03-27 22:22:34.875107411 -0500
+++ today/tmp/combined	2020-03-27 22:29:44.301053246 -0500
@@ -1,5 +1,5 @@
---- /home/jasonhu/Desktop/pydd/demodir/yesterday/patches/expcp/c	2020-03-27 22:18:39.761728051 -0500
-+++ /home/jasonhu/Desktop/pydd/demodir/yesterday/patches/expcp2/c	2020-03-27 14:30:08.494322171 -0500
+--- /home/jasonhu/Desktop/pydd/demodir/today/patches/expcp/c	2020-03-27 19:16:53.766614000 -0500
++++ /home/jasonhu/Desktop/pydd/demodir/today/patches/expcp2/c	2020-03-27 19:16:53.766614000 -0500
 @@ -8,13 +8,5 @@
  123123123123
  12312312asdfasdfasdf
diff -u -r yesterday/tmp/minimal_patch today/tmp/minimal_patch
